<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Batch Correction on SingleCellExperiment Object • singleCellTK</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Batch Correction on SingleCellExperiment Object">
<meta property="og:description" content="singleCellTK">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">singleCellTK</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.9.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/introduction_to_SCTK.html">General Workflow</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Shell Command Line Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/run_qc.html">Quality Control</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    R Console Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/import_scRNAseq_data_as_SCE.html">Import data into SCTK</a>
    </li>
    <li>
      <a href="../articles/run_qc.html">Quality Control</a>
    </li>
    <li>
      <a href="../articles/normalization.html">Normalization</a>
    </li>
    <li>
      <a href="../articles/batch_correction.html">Batch Correction</a>
    </li>
    <li>
      <a href="../articles/variability.html">Variability</a>
    </li>
    <li>
      <a href="../articles/feature_selection.html">Feature Selection</a>
    </li>
    <li>
      <a href="../articles/differential_expression.html">Differential Expression</a>
    </li>
    <li>
      <a href="../articles/find_marker.html">Find Marker via DE</a>
    </li>
    <li>
      <a href="../articles/visualize_results.html">Visualize Results</a>
    </li>
    <li>
      <a href="../articles/heatmap.html">Heatmap Visualization</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Shiny UI Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/ui_batch_correction.html">Batch Correction</a>
    </li>
    <li>
      <a href="../articles/ui_clustering.html">Clustering</a>
    </li>
    <li>
      <a href="../articles/ui_differential_expression.html">Differential Expression</a>
    </li>
    <li>
      <a href="../articles/ui_find_marker.html">Find Marker</a>
    </li>
    <li>
      <a href="../articles/ui_heatmap.html">Generic Heatmap</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/compbiomed/singleCellTK">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="batch_correction_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Batch Correction on SingleCellExperiment Object</h1>
                        <h4 class="author">Yichen Wang</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/compbiomed/singleCellTK/blob/master/vignettes/batch_correction.Rmd"><code>vignettes/batch_correction.Rmd</code></a></small>
      <div class="hidden name"><code>batch_correction.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h2>
<p>For this section, we wrapped 12 methods into functions that perform batch effect correction on an <a href="https://rdrr.io/bioc/SingleCellExperiment/man/SingleCellExperiment.html"><code>SingleCellExperiment</code></a> (SCE) object and return the input SCE object with a corrected matrix updated in-place.</p>
<p>Here is a table for the method names, citations, corresponding wrapper functions and where the results are updated:</p>
<table class="table">
<tr>
<th>
Method
</th>
<th>
Citation
</th>
<th>
Function
</th>
<th>
Result Slots
</th>
</tr>
<tr>
<td>
BBKNN
</td>
<td>
Polański, Krzysztof and et al., 2019
</td>
<td>
<a href="../reference/runBBKNN.html">runBBKNN()</a>
</td>
<td>
<code>reducedDim</code>
</td>
</tr>
<tr>
<td>
ComBat
</td>
<td>
Yuqing Zhang and et al., 2018
</td>
<td>
<a href="../reference/runComBat.html">runComBat()</a>
</td>
<td>
<code>assay</code>
</td>
</tr>
<tr>
<td>
FastMNN
</td>
<td>
Aaron Lun, 2018
</td>
<td>
<a href="../reference/runFastMNN.html">runFastMNN()</a>
</td>
<td>
<code>reducedDim</code>
</td>
</tr>
<tr>
<td>
MNN
</td>
<td>
Laleh Haghverdi, 2018
</td>
<td>
<a href="../reference/runMNNCorrect.html">runMNNCorrect()</a>
</td>
<td>
<code>assay</code>
</td>
</tr>
<tr>
<td>
Harmony
</td>
<td>
Ilya Korsunsky and et al., 2019
</td>
<td>
<a href="../reference/runHarmony.html">runHarmony()</a>
</td>
<td>
<code>reducedDim</code>
</td>
</tr>
<tr>
<td>
LIGER
</td>
<td>
Joshua Welch, et al., 2018
</td>
<td>
<a href="../reference/runLIGER.html">runLIGER()</a>
</td>
<td>
<code>reducedDim</code>
</td>
</tr>
<tr>
<td>
Limma
</td>
<td>
Gordon K Smyth, et al., 2003
</td>
<td>
<a href="../reference/runLimmaBC.html">runLimmaBC()</a>
</td>
<td>
<code>assay</code>
</td>
</tr>
<tr>
<td>
Scanorama
</td>
<td>
Brian Hie et al, 2019
</td>
<td>
<a href="../reference/runSCANORAMA.html">runSCANORAMA()</a>
</td>
<td>
<code>assay</code>
</td>
</tr>
<tr>
<td>
scGen
</td>
<td>
Mohammad Lotfollah et al., 2019
</td>
<td>
<a href="../reference/runSCGEN.html">runSCGEN()</a>
</td>
<td>
<code>assay</code>
</td>
</tr>
<tr>
<td>
scMerge
</td>
<td>
Yingxin Lin et al., 2019
</td>
<td>
<a href="../reference/runSCMerge.html">runSCMerge()</a>
</td>
<td>
<code>assay</code>
</td>
</tr>
<tr>
<td>
Seurat Integration
</td>
<td>
Tim Stuart et al., 2019
</td>
<td>
<a href="../reference/runSeurat3Integration.html">runSeurat3Integration()</a>
</td>
<td>
<code>altExp</code>
</td>
</tr>
<tr>
<td>
ZINBWaVE
</td>
<td>
Davide Risso et al., 2018
</td>
<td>
<a href="../reference/runZINBWaVE.html">runZINBWaVE()</a>
</td>
<td>
<code>reducedDim</code>
</td>
</tr>
</table>
<p>For a result returned in <code>reducedDim</code>, it means that some sorts of dimension reduction method, such as PCA and UMAP, is performed during the correction. For a result returned in <code>assay</code>, all of the original features (i.e. genes) remain the same, thus it is a full-sized assay. For a result returned in <code>altExp</code>, it means that the values in the corrected assay still stand for the expression level of the original features, instead of any low-dimension embedding, but the number of these features are less than original due to potential subsetting steps in the calculation.</p>
</div>
<div id="specifications" class="section level2">
<h2 class="hasAnchor">
<a href="#specifications" class="anchor"></a>Specifications</h2>
<ul>
<li>The pipeline is currently written in the R language. Users will need to install R version 3.6.2 (or higher) in order to run all of the required packages.</li>
<li>For using method “BBKNN”, “Scanorama” and “scGen”, the corresponding Python3 modules has to be installed</li>
</ul>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1"></a><span class="ex">pip</span> install scanpy bbknn scanorama scgen</span></code></pre></div>
</div>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p>The script will automatically try to install the “singleCellTK” package from Bioconductor if not available. However, currently this code is only located on a development branch which needs to be installed from GitHub:</p>
<div class="sourceCode" id="cb2"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">devtools</span>)
<span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span>(<span class="st">"compbiomed/singleCellTK@devel"</span>)</pre></body></html></div>
</div>
<div id="running-the-pipeline" class="section level2">
<h2 class="hasAnchor">
<a href="#running-the-pipeline" class="anchor"></a>Running the pipeline</h2>
<p>To run the pipeline, the most basic requirements are:</p>
<ul>
<li>An assay of expression value, usually pre-processed.</li>
<li>The annotation of the batches.</li>
</ul>
<p>As we adopt <a href="https://rdrr.io/bioc/SingleCellExperiment/"><code>SingleCellExperiment</code></a> (<code>sce</code>) object through out the whole SCTK for a collection of all matrices and metadatas of the dataset of interests, the assay to be corrected, called <code>"assayToCorr"</code>, has to be saved at <code>assay(sce, "assayToCorr")</code>. Meanwhile, the batch annotation information has to be saved in a column of <code>colData(sce)</code>.</p>
<blockquote>
<p>Note that the batch annotation should better be saved as a <code>factor</code> in the <code>colData</code>, especially when the batches are represented by integer numbers, because some downstream analysis are likely to parse the non-character and non-logical information as continuous values instead of categorical values.</p>
</blockquote>
<div id="command-line-example" class="section level3">
<h3 class="hasAnchor">
<a href="#command-line-example" class="anchor"></a>Command line example</h3>
<div id="prepare-an-sce-object-with-multiple-batches" class="section level4">
<h4 class="hasAnchor">
<a href="#prepare-an-sce-object-with-multiple-batches" class="anchor"></a>Prepare an SCE object with multiple batches</h4>
<p>Here we present an example dataset that is combined from “pbmc3k” and “pbmc4k” (Kasper D. Hansen <em>et al.</em>, 2017), which you can import by a function called <code><a href="../reference/importExampleData.html">importExampleData()</a></code>.</p>
<div class="sourceCode" id="cb3"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">singleCellTK</span>)
<span class="no">pbmc6k</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/importExampleData.html">importExampleData</a></span>(<span class="st">'pbmc6k'</span>)
<span class="no">pbmc8k</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/importExampleData.html">importExampleData</a></span>(<span class="st">'pbmc8k'</span>)
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span>(<span class="no">pbmc6k</span>), <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">'genes'</span>, <span class="st">'cells'</span>)))
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span>(<span class="no">pbmc8k</span>), <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">'genes'</span>, <span class="st">'cells'</span>)))</pre></body></html></div>
<pre><code>## [1] "32738 genes" "5419 cells" 
## [1] "33694 genes" "8381 cells"</code></pre>
<p>There is a function called <code><a href="../reference/combineSCE.html">combineSCE()</a></code>, which accepts a list of SCE objects as input and returns a combined SCE object. This function requires that the number of genes in each SCE object has to be the same, and the gene metadata (i.e. <code>rowData</code>) has to match with each other if the same fields exist. Therefore, we need some pre-process for the combination. Users do not necessarily have to follow the same way, depending on how the raw datasets are provided.</p>
<div class="sourceCode" id="cb5"><html><body><pre class="r"><span class="co">## The strategy here is to only use the genes presented in both datasets.</span>
<span class="co">## Get the intersection of both gene lists.</span>
<span class="no">isecGene</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sets.html">intersect</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">pbmc6k</span>), <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">pbmc8k</span>))
<span class="co">## Subset each dataset by the intersection.</span>
<span class="no">pbmc6k.isec</span> <span class="kw">&lt;-</span> <span class="no">pbmc6k</span>[<span class="no">isecGene</span>,]
<span class="no">pbmc8k.isec</span> <span class="kw">&lt;-</span> <span class="no">pbmc8k</span>[<span class="no">isecGene</span>,]
<span class="co">## Overwrite the mismatches in rowData, a special case in this example.</span>
<span class="fu">rowData</span>(<span class="no">pbmc8k.isec</span>) <span class="kw">&lt;-</span> <span class="fu">rowData</span>(<span class="no">pbmc6k.isec</span>)
<span class="co">## Apply combination function</span>
<span class="no">sce.combine</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/combineSCE.html">combineSCE</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="kw">pbmc6k</span> <span class="kw">=</span> <span class="no">pbmc6k.isec</span>,
                               <span class="kw">pbmc8k</span> <span class="kw">=</span> <span class="no">pbmc8k.isec</span>))
<span class="co">## Simply filter out the genes that are expressed in less than 1% of all cells.</span>
<span class="no">sce.filter</span> <span class="kw">&lt;-</span> <span class="no">sce.combine</span>[<span class="fu"><a href="https://rdrr.io/r/base/colSums.html">rowSums</a></span>(<span class="fu">assay</span>(<span class="no">sce.combine</span>) <span class="kw">&gt;</span> <span class="fl">0</span>) <span class="kw">&gt;=</span> <span class="fl">0.01</span> * <span class="fu"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span>(<span class="no">sce.combine</span>),]
<span class="no">sce.filter</span></pre></body></html></div>
<pre><code>## class: SingleCellExperiment 
## dim: 10020 13800 
## metadata(0):
## assays(1): counts
## rownames(10020): AP006222.2 RP11-206L10.9 ... MT-CYB AL592183.1
## rowData names(3): ENSEMBL_ID Symbol_TENx Symbol
## colnames(13800): pbmc6k_AAACATACAACCAC-1 pbmc6k_AAACATACACCAGT-1 ...
##   pbmc8k_TTTGTCATCTCGAGTA-1 pbmc8k_TTTGTCATCTGCTTGC-1
## colData names(12): Sample Barcode ... Date_published sample
## reducedDimNames(0):
## spikeNames(0):
## altExpNames(0):</code></pre>
<p>The batch annotation now is stored at <code>colData(sce.filter)$sample</code>. The summary can be checked by running:</p>
<div class="sourceCode" id="cb7"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span>(<span class="fu">colData</span>(<span class="no">sce.filter</span>)$<span class="no">sample</span>)</pre></body></html></div>
<pre><code>## 
## pbmc6k pbmc8k 
##   5419   8381</code></pre>
<p>Before trying out any batch correction function, we recommend to use a small subset of example so the effect can be quickly examined. Additionally, most of the batch correction methods provided require a log-normalized assay as input, so we have to make sure of this.</p>
<div class="sourceCode" id="cb9"><html><body><pre class="r"><span class="no">sce.small</span> <span class="kw">&lt;-</span> <span class="no">sce.filter</span>[<span class="fu"><a href="https://rdrr.io/r/base/sample.html">sample</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span>(<span class="no">sce.filter</span>), <span class="fl">800</span>), <span class="fu"><a href="https://rdrr.io/r/base/sample.html">sample</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span>(<span class="no">sce.filter</span>), <span class="fl">800</span>)]
<span class="no">sce.small</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/scater_logNormCounts.html">scater_logNormCounts</a></span>(<span class="kw">inSCE</span> <span class="kw">=</span> <span class="no">sce.small</span>,
                                  <span class="kw">useAssay</span> <span class="kw">=</span> <span class="st">'counts'</span>,
                                  <span class="kw">logAssayName</span> <span class="kw">=</span> <span class="st">'logcounts'</span>)
<span class="no">sce.small</span></pre></body></html></div>
<pre><code>## class: SingleCellExperiment 
## dim: 800 800 
## metadata(0):
## assays(2): counts logcounts
## rownames(800): RNF130 CERS4 ... ILF3 HEXB
## rowData names(3): ENSEMBL_ID Symbol_TENx Symbol
## colnames(800): pbmc8k_CGGACACCAAATTGCC-1 pbmc6k_CACTTTGACAGGAG-1 ...
##   pbmc8k_GCTTGAAGTCTTGTCC-1 pbmc6k_ATGATATGGACTAC-1
## colData names(12): Sample Barcode ... Date_published sample
## reducedDimNames(0):
## spikeNames(0):
## altExpNames(0):</code></pre>
</div>
<div id="run-a-batch-correction-method-on-the-prepared-sce-object" class="section level4">
<h4 class="hasAnchor">
<a href="#run-a-batch-correction-method-on-the-prepared-sce-object" class="anchor"></a>Run a batch correction method on the prepared SCE object</h4>
<p>The basic way to run a batch correction method from SingleCellTK is to select a function for the corresponding method, input the SCE object, specify the assay to correct, and the batch annotation.</p>
<p>For example, here we will try the batch correction method provided by <a href="https://rdrr.io/bioc/limma/man/removeBatchEffect.html">Limma</a>, which fits a linear model to the data.</p>
<div class="sourceCode" id="cb11"><html><body><pre class="r"><span class="no">sce.small</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/runLimmaBC.html">runLimmaBC</a></span>(<span class="kw">inSCE</span> <span class="kw">=</span> <span class="no">sce.small</span>,
                        <span class="kw">useAssay</span> <span class="kw">=</span> <span class="st">'logcounts'</span>,
                        <span class="kw">batch</span> <span class="kw">=</span> <span class="st">'sample'</span>,
                        <span class="kw">assayName</span> <span class="kw">=</span> <span class="st">'LIMMA'</span>)
<span class="no">sce.small</span></pre></body></html></div>
<pre><code>## class: SingleCellExperiment 
## dim: 800 800 
## metadata(0):
## assays(3): counts logcounts LIMMA
## rownames(800): RNF130 CERS4 ... ILF3 HEXB
## rowData names(3): ENSEMBL_ID Symbol_TENx Symbol
## colnames(800): pbmc8k_CGGACACCAAATTGCC-1 pbmc6k_CACTTTGACAGGAG-1 ...
##   pbmc8k_GCTTGAAGTCTTGTCC-1 pbmc6k_ATGATATGGACTAC-1
## colData names(12): Sample Barcode ... Date_published sample
## reducedDimNames(0):
## spikeNames(0):
## altExpNames(0):</code></pre>
</div>
<div id="visualization" class="section level4">
<h4 class="hasAnchor">
<a href="#visualization" class="anchor"></a>Visualization</h4>
<p>In this documentation, we provide three ways to examine the removal of batch effect, in terms of visualization.</p>
<ol style="list-style-type: decimal">
<li>Plot the variation explained by the batch annotation and another condition.</li>
</ol>
<p>This functionality is implemented in <code><a href="../reference/plotBatchVariance.html">plotBatchVariance()</a></code>. It plots a violin plot of the variation explained by the given batch annotation, an additional condition, and the variation explained by combining these two conditions.<br>
This plot would be useful for examining whether the existing batch effect is different than another condition (e.g. subtype) or is confounded by that. However, the additional condition labels (e.g. cell types) do not necessarily exist when batch effect removal is wanted, so only plotting the variation explained by batches is also supported.</p>
<div class="sourceCode" id="cb13"><html><body><pre class="r"><span class="fu"><a href="../reference/plotBatchVariance.html">plotBatchVariance</a></span>(<span class="kw">inSCE</span> <span class="kw">=</span> <span class="no">sce.small</span>,
                  <span class="kw">useAssay</span> <span class="kw">=</span> <span class="st">'logcounts'</span>,
                  <span class="kw">batch</span> <span class="kw">=</span> <span class="st">'sample'</span>,
                  <span class="kw">title</span> <span class="kw">=</span> <span class="st">'Variation Before Correction'</span>)</pre></body></html></div>
<p><img src="batch_correction_files/figure-html/batchVarBefore-1.png" width="700"></p>
<div class="sourceCode" id="cb14"><html><body><pre class="r"><span class="fu"><a href="../reference/plotBatchVariance.html">plotBatchVariance</a></span>(<span class="kw">inSCE</span> <span class="kw">=</span> <span class="no">sce.small</span>,
                  <span class="kw">useAssay</span> <span class="kw">=</span> <span class="st">'LIMMA'</span>,
                  <span class="kw">batch</span> <span class="kw">=</span> <span class="st">'sample'</span>,
                  <span class="kw">title</span> <span class="kw">=</span> <span class="st">'Variation After Correction'</span>)</pre></body></html></div>
<p><img src="batch_correction_files/figure-html/batchVarAfter-1.png" width="700"></p>
<ol start="2" style="list-style-type: decimal">
<li>Plot the mean expression level of each gene separately for each batch.</li>
</ol>
<p>This functionality is implemented in <code><a href="../reference/plotSCEBatchFeatureMean.html">plotSCEBatchFeatureMean()</a></code>. The methodology is straight forward, which plots violin plots for all the batches, and within each batch, the plot illustrates the distribution of the mean expression level of each gene. Thus the batch effect can be observed from the mean and standard deviation of each batch.</p>
<ol start="3" style="list-style-type: decimal">
<li>Plot a dimension reduced components to see the grouping of cells</li>
</ol>
<p>There is no function special for batch correction, but this can be achieved simply by using the dimension reduction calculation functions (e.g. <code><a href="../reference/getPCA.html">getPCA()</a></code>, <code><a href="../reference/getUMAP.html">getUMAP()</a></code> and <code><a href="../reference/getTSNE.html">getTSNE()</a></code>) and <code><a href="../reference/plotSCEDimReduceColData.html">plotSCEDimReduceColData()</a></code>.</p>
<div class="sourceCode" id="cb15"><html><body><pre class="r"><span class="no">sce.small</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/getPCA.html">getPCA</a></span>(<span class="kw">inSCE</span> <span class="kw">=</span> <span class="no">sce.small</span>,
                     <span class="kw">useAssay</span> <span class="kw">=</span> <span class="st">'logcounts'</span>,
                     <span class="kw">reducedDimName</span> <span class="kw">=</span> <span class="st">'PCA'</span>)
<span class="fu"><a href="../reference/plotSCEDimReduceColData.html">plotSCEDimReduceColData</a></span>(<span class="kw">inSCE</span> <span class="kw">=</span> <span class="no">sce.small</span>,
                        <span class="kw">colorBy</span> <span class="kw">=</span> <span class="st">'sample'</span>,
                        <span class="kw">reducedDimName</span> <span class="kw">=</span> <span class="st">'PCA'</span>,
                        <span class="kw">title</span> <span class="kw">=</span> <span class="st">'PCA Before Correction'</span>)</pre></body></html></div>
<p><img src="batch_correction_files/figure-html/umapBefore-1.png" width="700"></p>
<div class="sourceCode" id="cb16"><html><body><pre class="r"><span class="no">sce.small</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/getPCA.html">getPCA</a></span>(<span class="kw">inSCE</span> <span class="kw">=</span> <span class="no">sce.small</span>,
                     <span class="kw">useAssay</span> <span class="kw">=</span> <span class="st">'LIMMA'</span>,
                     <span class="kw">reducedDimName</span> <span class="kw">=</span> <span class="st">'PCA_LIMMA'</span>)
<span class="fu"><a href="../reference/plotSCEDimReduceColData.html">plotSCEDimReduceColData</a></span>(<span class="kw">inSCE</span> <span class="kw">=</span> <span class="no">sce.small</span>,
                        <span class="kw">colorBy</span> <span class="kw">=</span> <span class="st">'sample'</span>,
                        <span class="kw">reducedDimName</span> <span class="kw">=</span> <span class="st">'PCA_LIMMA'</span>,
                        <span class="kw">title</span> <span class="kw">=</span> <span class="st">'PCA After Correction'</span>)</pre></body></html></div>
<p><img src="batch_correction_files/figure-html/umapAfter-1.png" width="700"></p>
</div>
</div>
<div id="ui-usage" class="section level3">
<h3 class="hasAnchor">
<a href="#ui-usage" class="anchor"></a>UI usage</h3>
<p>The batch correction functionality is implemented under the tab “Normalization &amp; Batch Correction”, as shown in figure below, which is a sidebar layout.<br>
The correction is done with the settings in the sidebar on the left side. Users have to choose the assay to correct, the annotation for the batches, and the method to invoke. Below the method selection, users will be given alternative parameters specific to the method selected. The last text input is always for the name of the result, and its label indicates whether the result is updated as <code>reducedDim</code>, <code>assay</code>, or <code>altExp</code>. Duplicated names will automatically be overwritten without warning. After confirming all parameters, press “Run” to run the correction.<br>
The visualization check is implemented in the main panel on the right, with four plotting region and a list of settings. The plots are done in the exactly same way as Method 2 and 3 described above. After choosing the original assay used, batch annotation used, and the name of the result to check, press “Plot” to update the visualization.</p>
<p><img src="./batchCorrectionPanel.png"></p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Joshua D. Campbell, David Jenkins, Tyler Faits, Mohammed Muzamil Khan, Zhe Wang, Emma Briars, Sebastian Carrasco Pro, W. Evan Johnson, Yusuke Koga, Shruthi Bandyadka, Anastasia Leshchyk, Rui Hong, Yichen Wang.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
