<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Differential Expression • singleCellTK</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Differential Expression">
<meta property="og:description" content="singleCellTK">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">singleCellTK</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.9.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/introduction_to_SCTK.html">General Workflow</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Shell Command Line Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/run_qc.html">Quality Control</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    R Console Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/import_scRNAseq_data_as_SCE.html">Import data into SCTK</a>
    </li>
    <li>
      <a href="../articles/run_qc.html">Quality Control</a>
    </li>
    <li>
      <a href="../articles/normalization.html">Normalization</a>
    </li>
    <li>
      <a href="../articles/batch_correction.html">Batch Correction</a>
    </li>
    <li>
      <a href="../articles/variability.html">Variability</a>
    </li>
    <li>
      <a href="../articles/feature_selection.html">Feature Selection</a>
    </li>
    <li>
      <a href="../articles/differential_expression.html">Differential Expression</a>
    </li>
    <li>
      <a href="../articles/find_marker.html">Find Marker via DE</a>
    </li>
    <li>
      <a href="../articles/visualize_results.html">Visualize Results</a>
    </li>
    <li>
      <a href="../articles/heatmap.html">Heatmap Visualization</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Shiny UI Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/ui_batch_correction.html">Batch Correction</a>
    </li>
    <li>
      <a href="../articles/ui_clustering.html">Clustering</a>
    </li>
    <li>
      <a href="../articles/ui_differential_expression.html">Differential Expression</a>
    </li>
    <li>
      <a href="../articles/ui_find_marker.html">Find Marker</a>
    </li>
    <li>
      <a href="../articles/ui_heatmap.html">Generic Heatmap</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/compbiomed/singleCellTK">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="differential_expression_files/accessible-code-block-0.0.1/empty-anchor.js"></script><script src="differential_expression_files/kePrint-0.0.1/kePrint.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Differential Expression</h1>
                        <h4 class="author">Yichen Wang</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/compbiomed/singleCellTK/blob/master/vignettes/differential_expression.Rmd"><code>vignettes/differential_expression.Rmd</code></a></small>
      <div class="hidden name"><code>differential_expression.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h2>
<p>For this section, we provide wrappers of four methods, listed in the table below. All functions takes an input <code>SingleCellExperiment</code> (SCE) object and other settings as arguments and returns the same SCE object with results updated in <code>metadata</code> slot.</p>
<table class="table">
<thead><tr class="header">
<th align="center">Method</th>
<th align="left">Citation</th>
<th align="center">Function</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="center">MAST</td>
<td align="left">Greg Finak and et al., 2015</td>
<td align="center"><code><a href="../reference/runMAST.html">runMAST()</a></code></td>
</tr>
<tr class="even">
<td align="center">Limma</td>
<td align="left">Gordon Smyth and et al., 2004</td>
<td align="center"><code><a href="../reference/runLimmaDE.html">runLimmaDE()</a></code></td>
</tr>
<tr class="odd">
<td align="center">DESeq2</td>
<td align="left">Michael Love and et al., 2014</td>
<td align="center"><code><a href="../reference/runDESeq2.html">runDESeq2()</a></code></td>
</tr>
<tr class="even">
<td align="center">ANOVA</td>
<td align="left">Jeffrey T. Leek and et al., 2020</td>
<td align="center"><code><a href="../reference/runANOVA.html">runANOVA()</a></code></td>
</tr>
</tbody>
</table>
<p>A generic wrapper of all four methods is also provided, called <code><a href="../reference/runDEAnalysis.html">runDEAnalysis()</a></code>.</p>
</div>
<div id="workflow" class="section level2">
<h2 class="hasAnchor">
<a href="#workflow" class="anchor"></a>Workflow</h2>
<p>The differential expression analysis can be easily performed on any preprocessed SCE dataset. Here we introduce the workflow of using the generic wrapper <code><a href="../reference/runDEAnalysis.html">runDEAnalysis()</a></code>.</p>
<div id="parameters" class="section level3">
<h3 class="hasAnchor">
<a href="#parameters" class="anchor"></a>Parameters</h3>
<p>The most basic parameters include:</p>
<ul>
<li>The input SCE object - <code>inSCE</code>.</li>
<li>The method listed in the table above to use - <code>method</code>.</li>
<li>The assay to perform the analysis on - <code>useAssay</code>
</li>
</ul>
<p>Besides these, there are also a few other required parameter sets:</p>
<div id="conditions" class="section level4">
<h4 class="hasAnchor">
<a href="#conditions" class="anchor"></a>Conditions</h4>
<p>The R script functions we have allow users to perform differential expression analysis with relatively flexible condition setting. In terms of the condition of interests and the condition for control, comparison groups can be set by giving either one or more categories under a column in <code>colData</code>, or a prepared index vector as long as the indices are able to subset the input SCE object.</p>
<p>Additionally, only specifying the condition of interests is allowed and the control will then be set to all the other cells, which turns to a biomarker finding analysis.</p>
<p>Since the conditions can be set by using indices or annotations, we have two groups of parameters listed below. Note that only one way of setting can be used at one time.</p>
<ul>
<li>Use <code>index1</code> and <code>index2</code> for index style setting</li>
<li>Use <code>class</code> to specify the annotation vector by either directly giving a vector of proper length or giving a column name of <code>colData</code>. Then use <code>classGroup1</code>, <code>classGroup2</code> to specify which categories in <code>class</code> are of interests.</li>
</ul>
</div>
<div id="namings" class="section level4">
<h4 class="hasAnchor">
<a href="#namings" class="anchor"></a>Namings</h4>
<p>Something that might be special in our workflow is that, users will be required to specify name strings for:</p>
<ul>
<li>The group of interests - <code>groupName1</code>.</li>
<li>The group of control - <code>groupName2</code>.</li>
<li>The analysis run itself - <code>analysisName</code>.</li>
</ul>
<p>The reason is we assume that users are likely to perform multiple batches of analysis for a single dataset (e.g. group1 Vs. group2 and then group1 Vs. group3), and we hope to have everything stored in one SCE without leaving users any confusion when they look back on it. Also, the namings are also used by DE analysis related plotting functions, so that legends can be well annotated.</p>
</div>
</div>
<div id="example" class="section level3">
<h3 class="hasAnchor">
<a href="#example" class="anchor"></a>Example</h3>
<div id="preprocessing" class="section level4">
<h4 class="hasAnchor">
<a href="#preprocessing" class="anchor"></a>Preprocessing</h4>
<p>To demonstrate a simple and clear example, here we use the “PBMC-3k” dataset from “10X” which can be easily imported with SCTK functions. The preprocessing only includes necessary steps to get cluster labels (i.e. QC and filtering are excluded).</p>
<div class="sourceCode" id="cb1"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">singleCellTK</span>)
<span class="no">pbmc3k</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/importExampleData.html">importExampleData</a></span>(<span class="st">"pbmc3k"</span>)
<span class="no">pbmc3k</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/scater_logNormCounts.html">scater_logNormCounts</a></span>(<span class="no">pbmc3k</span>, <span class="st">"logcounts"</span>)
<span class="co"># Go through the Seurat curated workflow to get basic clusters</span>
<span class="no">pbmc3k</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/seuratNormalizeData.html">seuratNormalizeData</a></span>(<span class="kw">inSCE</span> <span class="kw">=</span> <span class="no">pbmc3k</span>, <span class="kw">useAssay</span> <span class="kw">=</span> <span class="st">"counts"</span>)
<span class="no">pbmc3k</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/seuratFindHVG.html">seuratFindHVG</a></span>(<span class="kw">inSCE</span> <span class="kw">=</span> <span class="no">pbmc3k</span>, <span class="kw">useAssay</span> <span class="kw">=</span> <span class="st">"seuratNormData"</span>)
<span class="no">pbmc3k</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/seuratScaleData.html">seuratScaleData</a></span>(<span class="kw">inSCE</span> <span class="kw">=</span> <span class="no">pbmc3k</span>, <span class="kw">useAssay</span> <span class="kw">=</span> <span class="st">"seuratNormData"</span>)
<span class="no">pbmc3k</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/seuratPCA.html">seuratPCA</a></span>(<span class="kw">inSCE</span> <span class="kw">=</span> <span class="no">pbmc3k</span>, <span class="kw">useAssay</span> <span class="kw">=</span> <span class="st">"seuratScaledData"</span>)
<span class="no">pbmc3k</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/seuratRunUMAP.html">seuratRunUMAP</a></span>(<span class="no">pbmc3k</span>)
<span class="no">pbmc3k</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/seuratFindClusters.html">seuratFindClusters</a></span>(<span class="kw">inSCE</span> <span class="kw">=</span> <span class="no">pbmc3k</span>, <span class="kw">useAssay</span> <span class="kw">=</span> <span class="st">"seuratScaledData"</span>)
<span class="co"># Optional visualization</span>
<span class="fu"><a href="../reference/plotSCEDimReduceColData.html">plotSCEDimReduceColData</a></span>(<span class="kw">inSCE</span> <span class="kw">=</span> <span class="no">pbmc3k</span>,
                        <span class="kw">colorBy</span> <span class="kw">=</span> <span class="st">"Seurat_louvain_Resolution0.8"</span>,
                        <span class="kw">conditionClass</span> <span class="kw">=</span> <span class="st">"factor"</span>,
                        <span class="kw">reducedDimName</span> <span class="kw">=</span> <span class="st">"seuratUMAP"</span>)</pre></body></html></div>
<p><img src="differential_expression_files/figure-html/de_prep-1.png" width="700" style="display: block; margin: auto;"></p>
</div>
<div id="run-differential-expression-analysis" class="section level4">
<h4 class="hasAnchor">
<a href="#run-differential-expression-analysis" class="anchor"></a>Run Differential Expression Analysis</h4>
<div class="sourceCode" id="cb2"><html><body><pre class="r"><span class="co"># Condition chosen basing on the UMAP</span>
<span class="no">pbmc3k</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/runDEAnalysis.html">runDEAnalysis</a></span>(<span class="kw">inSCE</span> <span class="kw">=</span> <span class="no">pbmc3k</span>,
                        <span class="kw">method</span> <span class="kw">=</span> <span class="st">"MAST"</span>,
                        <span class="kw">useAssay</span> <span class="kw">=</span> <span class="st">"logcounts"</span>,
                        <span class="kw">class</span> <span class="kw">=</span> <span class="st">"Seurat_louvain_Resolution0.8"</span>,
                        <span class="kw">classGroup1</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">2</span>), <span class="kw">classGroup2</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">4</span>, <span class="fl">5</span>),
                        <span class="kw">groupName1</span> <span class="kw">=</span> <span class="st">"c2"</span>, <span class="kw">groupName2</span> <span class="kw">=</span> <span class="st">"c4_c5"</span>,
                        <span class="kw">analysisName</span> <span class="kw">=</span> <span class="st">"c2_VS_c4_c5"</span>)</pre></body></html></div>
</div>
<div id="results" class="section level4">
<h4 class="hasAnchor">
<a href="#results" class="anchor"></a>Results</h4>
<p>The results are saved in the <code>metadata</code> slot of the returned SCE object, following the structure below:</p>
<pre><code>matadata(pbmc3k)
          |- $info1
          |- $info2
          |- ...
          |- $diffExp
                  |-$AnalysisName1
                  |-$AnalysisName2
                  |-...
                  |-$c2_VS_c3_c8
                             |-$useAssay    = "logcounts"
                             |-$groupNames  = c("c2", "c3_c8")
                             |-$select
                             |      |-$ix1  = c(FALSE, TRUE,  FALSE, ...)
                             |      |-$ix2  = c(FALSE, FALSE, FALSE, ...)
                             |
                             |        (Two logical vectors both having `ncol(inSCE)` values,
                             |         specifies which cells are selected for "c2" or "c3_c8")
                             |
                             |-$annotation  = "Seurat_louvain_Resolution0.8"
                             |-$result      = (the `data.frame` of top DEG table, shown below)
                             |-$method      = "MAST"</code></pre>
<p>To fetch the result as a table of the Top differential expressed genes and the statistics:</p>
<div class="sourceCode" id="cb4"><html><body><pre class="r"><span class="no">DEG</span> <span class="kw">&lt;-</span> <span class="fu">metadata</span>(<span class="no">pbmc3k</span>)$<span class="no">diffExp</span>$<span class="no">c2_VS_c4_c5</span>$<span class="no">result</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="no">DEG</span>)</pre></body></html></div>
<table class="table table table-striped table-hover">
<thead><tr>
<th style="text-align:left;">
Gene
</th>
<th style="text-align:right;">
Pvalue
</th>
<th style="text-align:right;">
Log2_FC
</th>
<th style="text-align:right;">
FDR
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
FTL
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
-3.202606
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
LYZ
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
-4.896037
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
FTH1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
-2.750537
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
S100A4
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
-3.673227
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
TYROBP
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
-3.532855
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
CST3
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
-3.612471
</td>
<td style="text-align:right;">
0
</td>
</tr>
</tbody>
</table>
<p>To visualize the result in a heatmap:</p>
<div class="sourceCode" id="cb5"><html><body><pre class="r"><span class="fu"><a href="../reference/plotDEGHeatmap.html">plotDEGHeatmap</a></span>(<span class="no">pbmc3k</span>, <span class="kw">useResult</span> <span class="kw">=</span> <span class="st">"c2_VS_c4_c5"</span>, <span class="kw">log2fcThreshold</span> <span class="kw">=</span> <span class="fl">1</span>)</pre></body></html></div>
<p><img src="differential_expression_files/figure-html/plotDE-1.png" width="864" style="display: block; margin: auto;"></p>
<p>For other approaches of visualizing top differential expressed genes, refer to <code><a href="../reference/plotDEGRegression.html">plotDEGRegression()</a></code> and <code><a href="../reference/plotDEGViolin.html">plotDEGViolin()</a></code>, which are borrowed from <a href="https://www.bioconductor.org/packages/release/bioc/vignettes/MAST/inst/doc/MAITAnalysis.html#401_visualization_of_50_most_differentially_expressed_genes">MAST’s tutorial</a>.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Joshua D. Campbell, David Jenkins, Tyler Faits, Mohammed Muzamil Khan, Zhe Wang, Emma Briars, Sebastian Carrasco Pro, W. Evan Johnson, Yusuke Koga, Shruthi Bandyadka, Anastasia Leshchyk, Rui Hong, Yichen Wang.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
