<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Find Marker - UI • singleCellTK</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Find Marker - UI">
<meta property="og:description" content="singleCellTK">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">singleCellTK</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.9.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/introduction_to_SCTK.html">General Workflow</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Shell Command Line Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/run_qc.html">Quality Control</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    R Console Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/import_scRNAseq_data_as_SCE.html">Import data into SCTK</a>
    </li>
    <li>
      <a href="../articles/run_qc.html">Quality Control</a>
    </li>
    <li>
      <a href="../articles/normalization.html">Normalization</a>
    </li>
    <li>
      <a href="../articles/batch_correction.html">Batch Correction</a>
    </li>
    <li>
      <a href="../articles/variability.html">Variability</a>
    </li>
    <li>
      <a href="../articles/feature_selection.html">Feature Selection</a>
    </li>
    <li>
      <a href="../articles/differential_expression.html">Differential Expression</a>
    </li>
    <li>
      <a href="../articles/find_marker.html">Find Marker via DE</a>
    </li>
    <li>
      <a href="../articles/visualize_results.html">Visualize Results</a>
    </li>
    <li>
      <a href="../articles/heatmap.html">Heatmap Visualization</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Shiny UI Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/ui_batch_correction.html">Batch Correction</a>
    </li>
    <li>
      <a href="../articles/ui_clustering.html">Clustering</a>
    </li>
    <li>
      <a href="../articles/ui_differential_expression.html">Differential Expression</a>
    </li>
    <li>
      <a href="../articles/ui_find_marker.html">Find Marker</a>
    </li>
    <li>
      <a href="../articles/ui_heatmap.html">Generic Heatmap</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/compbiomed/singleCellTK">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="ui_find_marker_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Find Marker - UI</h1>
                        <h4 class="author">Yichen Wang</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/compbiomed/singleCellTK/blob/master/vignettes/ui_find_marker.Rmd"><code>vignettes/ui_find_marker.Rmd</code></a></small>
      <div class="hidden name"><code>ui_find_marker.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h2>
<p>This section describes the usage of the user interface (UI) for singleCellTK (SCTK) marker finding workflow. The underneath process is wrapped by R function <code><a href="../reference/findMarkerDiffExp.html">findMarkerDiffExp()</a></code>. For the help of R console workflow, also refer to <a href="find_marker.html">the help page for R console</a>.</p>
</div>
<div id="ui-layout" class="section level2">
<h2 class="hasAnchor">
<a href="#ui-layout" class="anchor"></a>UI Layout</h2>
<div id="entry-of-the-panel" class="section level3">
<h3 class="hasAnchor">
<a href="#entry-of-the-panel" class="anchor"></a>Entry of The Panel</h3>
<p>From anywhere of the UI, the panel for marker finding can be accessed from the top navigation panel at the circled tab shown below.</p>
<p><img src="fm_ui_entry.png" alt="Entry"><br></p>
<p>The UI is constructed in a sidebar pattern. The sidebar is mainly for the parameters essential for running the marker finding algorithm. The main panel is mainly for demonstrating the results, including the marker table and the heatmap.</p>
</div>
<div id="sidebar---parameters" class="section level3">
<h3 class="hasAnchor">
<a href="#sidebar---parameters" class="anchor"></a>Sidebar - Parameters</h3>
<p><img src="fm_ui_main1.png" style="border: 1px solid rgba(0,0,0,0.2)"></p>
<ul>
<li>Assay selection - selection input <strong>“Select Assay”</strong>. Users will choose which expression data matrix to use for marker detection. Note that for method “MAST” and “Limma”, a logged count matrix is required.</li>
<li>Method selection - selection input <strong>“Select Differential Expression Method”</strong>. Users will choose which method to use. The method here means the intrinsic dependency used for differential expression (DE) analysis.</li>
<li>Cluster definition - selection input <strong>“Cluster Annotation”</strong>. Users will choose a cell level annotation, which defines the cluster assignment for each cell. The options come from the cell annotation (<code>colData</code>) in the background SingleCellExperiment (SCE) object.</li>
<li>Log2FC cutoff setting - numeric input <strong>“Log2FC greater than”</strong>. The cutoff set here will rule out DE genes with absolute value of Log2FC (logged fold change) smaller than the cutoff from the result.</li>
<li>FDR cutoff setting - numeric input <strong>“FDR less than”</strong>. The cutoff set here will rule out DE genes with FDR (false discovery rate) value more than the cutoff from the result.</li>
</ul>
</div>
<div id="main-panel---result-table" class="section level3">
<h3 class="hasAnchor">
<a href="#main-panel---result-table" class="anchor"></a>Main Panel - Result Table</h3>
<p>The table is shown in the figure above. In this tab, there will be a table for all the markers detected and passed the filters. It is constructed with 5 columns, including “Gene” - the default feature identifiers (not necessarily a gene ID or symbol, depending on the background SCE object); “Log2_FC”, “Pvalue” and “FDR” - the statistics that supports the detection; and the fifth one that labels which cluster the marker is for, and the column name is exactly the “cluster annotation” selected in the sidebar. Users can also download the table in comma-splitted value (CSV) format, by clicking <strong>“Download Result Table”</strong> button.</p>
</div>
<div id="main-panel---heatmap" class="section level3">
<h3 class="hasAnchor">
<a href="#main-panel---heatmap" class="anchor"></a>Main Panel - Heatmap</h3>
<p><img src="fm_ui_main2.png" style="border: 1px solid rgba(0,0,0,0.2)"></p>
<p>For the heatmap visualization, we have another inside sidebar for its setting.</p>
<ul>
<li>Assay selection - selection input <strong>“Assay to plot”</strong>. User will select the expression data matrix here for the heatmap plotting. The UI will automatically make a selection basing on the selected one in the left side bar when “Find Marker” button is pressed, but users are allowed to change to another option.</li>
<li>Cluster ordering - radio button <strong>“Order blocks by”</strong>. User will determine here whether to order the columns of the heatmap by the cluster name (i.e. annotation) or the size (i.e. number of cells). Checkbox <strong>“Decreasing”</strong>. Users can check this option for putting the alphabetically latter cluster or cluster with the largest size on the left.</li>
<li>Top N marker cutoff - checkbox and numeric input <strong>“Plot Top N markers of each cluster”</strong>. If the checkbox is not checked, all markers passing the running filter (mentioned above) and plotting filters (will mention below) will be plotted. Otherwise checked, only the top <code>N</code> markers, as <code>N</code> is the input numeric value in the numeric input below the checkbox, for each cluster will be plotted. The ranking is based on Log2FC value. When less than <code>N</code> markers are detected for one cluster, all of them will be plotted.</li>
<li>Log2FC cutoff - numeric input <strong>“Plot Log2FC greater than”</strong>. The cutoff set here will rule out markers with absolute value of Log2FC (logged fold change) smaller than the cutoff from the plotting. This will not affect the result saved when running the algorithm.</li>
<li>FDR cutoff - numeric input <strong>""</strong>. The cutoff set here will rule out markers with FDR (false discovery rate) value more than the cutoff from the plotting. Similarly, this will not affect the result saved when running the algorithm.</li>
<li>Additional annotation - multiple selection input <strong>“Additional feature annotation”</strong> and <strong>“Additional cell annotation”</strong>. User can add up more cell or gene annotation to the heatmap lagend.</li>
</ul>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Joshua D. Campbell, David Jenkins, Tyler Faits, Mohammed Muzamil Khan, Zhe Wang, Emma Briars, Sebastian Carrasco Pro, W. Evan Johnson, Yusuke Koga, Shruthi Bandyadka, Anastasia Leshchyk, Rui Hong, Yichen Wang.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
