<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Heatmap - UI • singleCellTK</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Heatmap - UI">
<meta property="og:description" content="singleCellTK">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">singleCellTK</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.9.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/introduction_to_SCTK.html">General Workflow</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Shell Command Line Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/run_qc.html">Quality Control</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    R Console Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/import_scRNAseq_data_as_SCE.html">Import data into SCTK</a>
    </li>
    <li>
      <a href="../articles/run_qc.html">Quality Control</a>
    </li>
    <li>
      <a href="../articles/normalization.html">Normalization</a>
    </li>
    <li>
      <a href="../articles/batch_correction.html">Batch Correction</a>
    </li>
    <li>
      <a href="../articles/variability.html">Variability</a>
    </li>
    <li>
      <a href="../articles/feature_selection.html">Feature Selection</a>
    </li>
    <li>
      <a href="../articles/differential_expression.html">Differential Expression</a>
    </li>
    <li>
      <a href="../articles/find_marker.html">Find Marker via DE</a>
    </li>
    <li>
      <a href="../articles/visualize_results.html">Visualize Results</a>
    </li>
    <li>
      <a href="../articles/heatmap.html">Heatmap Visualization</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Shiny UI Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/ui_batch_correction.html">Batch Correction</a>
    </li>
    <li>
      <a href="../articles/ui_clustering.html">Clustering</a>
    </li>
    <li>
      <a href="../articles/ui_differential_expression.html">Differential Expression</a>
    </li>
    <li>
      <a href="../articles/ui_find_marker.html">Find Marker</a>
    </li>
    <li>
      <a href="../articles/ui_heatmap.html">Generic Heatmap</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/compbiomed/singleCellTK">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="ui_heatmap_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Heatmap - UI</h1>
                        <h4 class="author">Yichen Wang</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/compbiomed/singleCellTK/blob/master/vignettes/ui_heatmap.Rmd"><code>vignettes/ui_heatmap.Rmd</code></a></small>
      <div class="hidden name"><code>ui_heatmap.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h2>
<p>This section describes the usage of the user interface (UI) for singleCellTK (SCTK) generic heatmap plotting. The underneath process is wrapped by R function <code><a href="../reference/plotSCEHeatmap.html">plotSCEHeatmap()</a></code>. For the help of R console workflow, also refer to <a href="heatmap.html">the help page for R console</a>.</p>
</div>
<div id="ui-layout" class="section level2">
<h2 class="hasAnchor">
<a href="#ui-layout" class="anchor"></a>UI Layout</h2>
<div class="figure">
<img src="hm_ui_entry.png" alt=""><p class="caption">entry</p>
</div>
<p>The UI for plotting a generic heatmap is constructed with 5 sections: data input, data subsetting, annotation adding, parameter setting, and finally, plotting.</p>
<div id="data-input" class="section level3">
<h3 class="hasAnchor">
<a href="#data-input" class="anchor"></a>Data Input</h3>
<p><img src="hm_ui_main1.png" style="border: 1px solid rgba(0,0,0,0.2)"></p>
<p>The data used for plotting a heatmap can only be a feature expression matrix, technically called <code>assay</code> in R language. This will be set at the selection input <strong>“Assay to Plot”</strong>.</p>
<p>There is a second selection called <strong>“Import from analysis”</strong> which is used for the fast setting of plotting a differential expression (DE) analysis or marker detection specific style. The detail of this feature will be introduced <a href="#additional-usage">later</a>.</p>
</div>
<div id="data-subsetting" class="section level3">
<h3 class="hasAnchor">
<a href="#data-subsetting" class="anchor"></a>Data Subsetting</h3>
<p><img src="hm_ui_main2_1.png" style="border: 1px solid rgba(0,0,0,0.2)"></p>
<p>Similar to the condition definition in the DE analysis panel, here SCTK also adopts a data table to do a filtering and select the cells and features, in order to maintain the maximum flexibility. By default, all the cell annotation available will be displayed as shown in the screenshot above, as well as the feature annotation.</p>
<p>Generally, to make a selection, users will first find the annotations which can define the cells or features of interests, apply the filter, and then press <strong>“Add all filtered”</strong> to finish the selection.</p>
<p>To make the operation cleaner, user can choose to only display the annotation classes of interests at the multi-selection input <strong>“Columns to display”</strong>. The outcome will look like the screenshot below.</p>
<p><img src="hm_ui_main2_1_2.png" style="border: 1px solid rgba(0,0,0,0.2)"></p>
<p>The screenshot above also demonstrates one type of annotation filtering - by category. When the annotation of interests is categorical (in terms of R language, a <code>character</code> or <code>logical</code> vector or a <code>factor</code>), filters can be applied by selecting the categories of interest. While the screenshots below demonstrates another type of filering - by limiting a range to a continuous data (in R, a <code>numeric</code> or <code>integer</code> vector).</p>
<p><img src="hm_ui_main2_1_3.png" style="border: 1px solid rgba(0,0,0,0.2)"></p>
<p>Finally, if no annotation is available, or if users are not sure what annotation to use, SCTK provides a third tab that allows direct pasting a list of unique cell/feature identifiers. As shown below:</p>
<p><img src="hm_ui_main2_2.png" style="border: 1px solid rgba(0,0,0,0.2)"></p>
<p>The identifiers pasted to the text box should be able to uniquely identify the cells or features, and they have to be able to be found in the background information (i.e. the background SingleCellExperiment [SCE] object). However, they do not necessarily have to be the default identifiers. That is to say, identifiers that can be found in existing annotation are also acceptable. To accomplish this, users only need to select what class of annotation they intend to use at the selection input <strong>“Match input cell/feature identifiers by”</strong>. Note that the option “Row Names” indicates the default identifier when no annotation is available. For example, usually in a “cellRanger” style data, the default feature identifiers would usually be “Ensembl IDs”, while “gene symbols” will be embedded in the feature annotation. When users only have a list of genes symbol, they can directly use this list, and select the annotation for “gene symbol”, instead of going for some other external tools to transfer the “symbols” into “IDs”.</p>
</div>
<div id="annotation-adding" class="section level3">
<h3 class="hasAnchor">
<a href="#annotation-adding" class="anchor"></a>Annotation Adding</h3>
<p><img src="hm_ui_main3_1.png" style="border: 1px solid rgba(0,0,0,0.2)"></p>
<p>In this sub-panel, users can attach the annotations to be shown in the heatmap legend, and changes in colors are allowed. As mentioned previously, there are two types of values - categorical and continuous. For categorical information, the color setting will look like the screenshot above. All categories will be displayed with a color input.</p>
<p><img src="hm_ui_main3_2.png" style="border: 1px solid rgba(0,0,0,0.2)"></p>
<p>While for a continuous value, users will still be asked if the value is categorical or continuous. The reason is that, technically, sometimes there are annotations generated in a continuous numeric data type but indicating categorical information, such as clusters labeled with integers. If users select “Categorical”, the similar color selections will be displayed as in the previous screenshot. If “Continuous” is selected, a different color selection UI will be shown, as the screenshot above. Here users will be asked to set two colors for a color gradient.</p>
<p>Note that when categorical annotation is wanted but too many categories are detected, SCTK will not display any color selection, in order to avoid a UI “overflow”.</p>
</div>
<div id="other-parameters" class="section level3">
<h3 class="hasAnchor">
<a href="#other-parameters" class="anchor"></a>Other Parameters</h3>
<p><img src="hm_ui_main4.png" style="border: 1px solid rgba(0,0,0,0.2)"></p>
<p>This is the last sub-panel for the generic heatmap setting.</p>
<ul>
<li>Heatmap splitting - multi-selection inputs <strong>“Split columns (cell) by”</strong> and <strong>“Split rows (feature) by”</strong>. By splitting the heatmap in row/column, the clustering of rows/columns will be performed in each splitted group separately. The procedure is also called semi-heatmap in some cases.</li>
<li>Label adding - checkbox inputs <strong>“Add cell/feature labels”</strong>. The labels added here are text identifiers attached next to each row/column. Not recommanded when there are a large amount of features/cells to plot.</li>
<li>Dendrogram adding - checkbox inputs <strong>“Show dendrograms for”</strong>. Adding the dendrogram of cell/feature clustering to the top/left. The dendrograms will also be splitted if the heatmap splitting is also requested.</li>
<li>Normalization - checkbox input <strong>“Z-Score SCale”</strong>. Whether to perform z-score normalization to the data matrix to plot.</li>
<li>Value trimming - numeric sliding input <strong>“Trim”</strong>. Trim the value in the data matrix to plot to the closer bound of a range if a value is out of the range. The numeric slider UI dynamically changes with the data if normalized.</li>
<li>Heatmap color setting - <strong>“Color Scheme”</strong>. User can set the colors of the highest value, the lowest value and the middle value for the heatmap. Meanwhile, some frequently used presets are also provided. s</li>
</ul>
</div>
<div id="plotting-region" class="section level3">
<h3 class="hasAnchor">
<a href="#plotting-region" class="anchor"></a>Plotting Region</h3>
<p><img src="hm_ui_plot1.png" style="border: 1px solid rgba(0,0,0,0.2)"></p>
<p>By applying all the settings shown in the screenshots of the previous sections, a heatmap in the screenshot above will be generated, by clicking on the button <strong>“Plot Heatmap”</strong>.</p>
</div>
</div>
<div id="additional-usage" class="section level2">
<h2 class="hasAnchor">
<a href="#additional-usage" class="anchor"></a>Additional Usage</h2>
<p>The documentation above is for the general usage of SCTK Heatmap page. While heatmap is an important visualization method for other analysis, SCTK allows importing some types analysis results. Currently, DE analysis and marker detection results are supported. Here, we will present an example of importing the DE analysis result, which was produced in the <a href="ui_differential_expression.html">DE UI documentation</a>.</p>
<p><img src="hm_ui_de_1.png" style="border: 1px solid rgba(0,0,0,0.2)"></p>
<p>First, users need to go back to the top of the Heatmap panel, and find <strong>“Import from analysis”</strong> selection input. On selecting <code>"Differential Expression"</code>, the session will scan the background data and show a list of the analysis information for users to choose from. After selecting the analysis of interests, click <strong>“Import”</strong>.</p>
<p>The import functionality will automatically fill most of the options afterwards, including making up some temporary DE specific annotations and subset the data accordingly.</p>
<p><img src="hm_ui_de_2.png" style="border: 1px solid rgba(0,0,0,0.2)"><img src="hm_ui_de_3.png" style="border: 1px solid rgba(0,0,0,0.2)"></p>
<p>Note that the features automatically selected are all of the ones stored to the result. In other words, this is affected by the parameters used when running the DE analysis. However, sometimes users might not want to plot all of them but apply some further filters instead. For example, using features with the absolute value of Log2FC greater than <code>1</code> usually produces cleaner figure, than setting that bound to <code>0.25</code>. To apply this filter in the UI, users need to click on the input box below the column title in the table. If the column is for categorical information, then just click on the categories of interests; else if it is numeric, users can either manually drag the sliders or type a “formula” into the input box. The formula follows the rule: <code>{low}...{high}</code>, where one of the two bounds can be omitted for a one-end range. For example if users want to apply a filter described above for Log2FC values, they need to first click <strong>“Clear selection”</strong>, type <code>1...</code> , click <strong>“Add all filtered”</strong>, then type <code>...-1</code>, and click <strong>“Add all filtered”</strong> again.</p>
<p><img src="hm_ui_de_8.png" style="border: 1px solid rgba(0,0,0,0.2)"></p>
<p>Other heatmap settings will also be automatically filled for a DE specific heatmap.</p>
<p><img src="hm_ui_de_4.png" style="border: 1px solid rgba(0,0,0,0.2)"><img src="hm_ui_de_5.png" style="border: 1px solid rgba(0,0,0,0.2)"><img src="hm_ui_de_6.png" style="border: 1px solid rgba(0,0,0,0.2)"><img src="hm_ui_de_7.png" style="border: 1px solid rgba(0,0,0,0.2)"></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Joshua D. Campbell, David Jenkins, Tyler Faits, Mohammed Muzamil Khan, Zhe Wang, Emma Briars, Sebastian Carrasco Pro, W. Evan Johnson, Yusuke Koga, Shruthi Bandyadka, Anastasia Leshchyk, Rui Hong, Yichen Wang.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
