<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Processing and Visualizing Data in the Single Cell Toolkit • singleCellTK</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootswatch/3.3.7/yeti/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Processing and Visualizing Data in the Single Cell Toolkit">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">singleCellTK</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">1.1.18</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/v01-Introduction_to_singleCellTK.html">Introduction to singleCellTK</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Interactive Shiny Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/v03-tab01_Upload.html">1: Upload</a>
    </li>
    <li>
      <a href="../articles/v04-tab02_Data-Summary-and-Filtering.html">2: Data Summary and Filtering</a>
    </li>
    <li>
      <a href="../articles/v05-tab03_Visualization-and-Clustering.html">3: Visualization and Clustering</a>
    </li>
    <li>
      <a href="../articles/v06-tab04_Batch-Correction.html">4: Batch Correction</a>
    </li>
    <li>
      <a href="../articles/v07-tab05_Differential-Expression.html">5: Differential Expression</a>
    </li>
    <li>
      <a href="../articles/v08-tab06_Pathway-Activity-Analysis.html">6: Pathway Activity Analysis</a>
    </li>
    <li>
      <a href="../articles/v09-tab07_Sample-Size.html">7: Sample Size</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Command Line Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/v02-Processing_and_Visualizing_Data_in_the_SingleCellTK.html">Processing and Visualizing Data</a>
    </li>
    <li>
      <a href="../articles/v10-Aligning_and_Quantifying_scRNA-Seq_Data.html">Aligning and Quantifying scRNA-Seq Data</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/compbiomed/singleCellTK">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Processing and Visualizing Data in the Single Cell Toolkit</h1>
                        <h4 class="author">David Jenkins</h4>
            <address class="author_afil">
      The Section of Computational Biomedicine, Boston University School of Medicine, Boston, MA<br><a class="author_email" href="mailto:#"></a><a href="mailto:dfj@bu.edu">dfj@bu.edu</a>
      </address>
                              <h4 class="author">Tyler Faits</h4>
            <address class="author_afil">
      The Section of Computational Biomedicine, Boston University School of Medicine, Boston, MA<br><h4 class="author">W. Evan Johnson</h4>
            <address class="author_afil">
      The Section of Computational Biomedicine, Boston University School of Medicine, Boston, MA<br><div class="hidden name"><code>v02-Processing_and_Visualizing_Data_in_the_SingleCellTK.Rmd</code></div>

    </address>
</address>
</div>

    
    
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>The analysis modules available through the Shiny app are also available as R functions for standard R console processing of single cell RNA-Seq data using a SCtkExperiment object. At any stage, you can load the Shiny App to interactively visualize and analyze a data set, but this vignette will show a standard workflow run entirely through the R console.</p>
</div>
<div id="maits-example" class="section level1">
<h1 class="hasAnchor">
<a href="#maits-example" class="anchor"></a>MAITS Example</h1>
<p>The MAST package contains a convenient scRNA-Seq example data set of 96 Mucosal Associated Invariant T cells (MAITs), half of which were stimulated with cytokines to induce a response. For more details, consult the <a href="https://www.bioconductor.org/packages/release/bioc/html/MAST.html">MAST package</a> and <a href="https://www.bioconductor.org/packages/release/bioc/vignettes/MAST/inst/doc/MAITAnalysis.html">vignette</a>.</p>
<p>We will first convert the MAST example dataset to a SCtkExperiment object.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">suppressPackageStartupMessages</span>({
  <span class="kw">library</span>(MAST)
  <span class="kw">library</span>(singleCellTK)
  <span class="kw">library</span>(xtable)
})

<span class="kw">data</span>(maits, <span class="dt">package=</span><span class="st">"MAST"</span>)
maits_sce &lt;-<span class="st"> </span><span class="kw"><a href="../reference/createSCE.html">createSCE</a></span>(<span class="dt">assayFile =</span> <span class="kw">t</span>(maits<span class="op">$</span>expressionmat),
                       <span class="dt">annotFile =</span> maits<span class="op">$</span>cdat,
                       <span class="dt">featureFile =</span> maits<span class="op">$</span>fdat,
                       <span class="dt">assayName =</span> <span class="st">"logtpm"</span>,
                       <span class="dt">inputDataFrames =</span> <span class="ot">TRUE</span>,
                       <span class="dt">createLogCounts =</span> <span class="ot">FALSE</span>)
<span class="kw">rm</span>(maits)</code></pre></div>
<div id="summarizetable" class="section level2">
<h2 class="hasAnchor">
<a href="#summarizetable" class="anchor"></a>summarizeTable</h2>
<p>You can get summary metrics with the <code>summarizeTable</code> function:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">knitr<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/knitr/topics/kable">kable</a></span>(<span class="kw"><a href="../reference/summarizeTable.html">summarizeTable</a></span>(maits_sce, <span class="dt">useAssay =</span> <span class="st">"logtpm"</span>))</code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left">Metric</th>
<th align="right">Value</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Number of Samples</td>
<td align="right">96</td>
</tr>
<tr class="even">
<td align="left">Number of Genes</td>
<td align="right">16302</td>
</tr>
<tr class="odd">
<td align="left">Average number of reads per cell</td>
<td align="right">17867</td>
</tr>
<tr class="even">
<td align="left">Average number of genes per cell</td>
<td align="right">6833</td>
</tr>
<tr class="odd">
<td align="left">Samples with &lt;1700 detected genes</td>
<td align="right">5</td>
</tr>
<tr class="even">
<td align="left">Genes with no expression across all samples</td>
<td align="right">0</td>
</tr>
</tbody>
</table>
<p>Typically, these summary statistics would be run on a “counts” matrix, but here we have log(tpm) values so the average number of reads per cell is calculated from the normalized values instead of raw counts.</p>
</div>
<div id="filtering-by-annotation" class="section level2">
<h2 class="hasAnchor">
<a href="#filtering-by-annotation" class="anchor"></a>Filtering by Annotation</h2>
<p>Explore the available annotations in the data:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">colnames</span>(<span class="kw">colData</span>(maits_sce))</code></pre></div>
<pre><code>##  [1] "wellKey"          "condition"        "nGeneOn"         
##  [4] "libSize"          "PercentToHuman"   "MedianCVCoverage"
##  [7] "PCRDuplicate"     "exonRate"         "pastFastqc"      
## [10] "ncells"           "ngeneson"         "cngeneson"       
## [13] "TRAV1"            "TRBV6"            "TRBV4"           
## [16] "TRBV20"           "alpha"            "beta"            
## [19] "ac"               "bc"               "ourfilter"</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">table</span>(<span class="kw">colData</span>(maits_sce)<span class="op">$</span>ourfilter)</code></pre></div>
<pre><code>## 
## FALSE  TRUE 
##    22    74</code></pre>
<p>The data has a filtered dataset with 74 ‘pass filter’ samples, let’s subset the data to include the pass filter samples</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">maits_subset &lt;-<span class="st"> </span>maits_sce[, <span class="kw">colData</span>(maits_sce)<span class="op">$</span>ourfilter]
<span class="kw">table</span>(<span class="kw">colData</span>(maits_subset)<span class="op">$</span>ourfilter)</code></pre></div>
<pre><code>## 
## TRUE 
##   74</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">knitr<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/knitr/topics/kable">kable</a></span>(<span class="kw"><a href="../reference/summarizeTable.html">summarizeTable</a></span>(maits_subset, <span class="dt">useAssay =</span> <span class="st">"logtpm"</span>))</code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left">Metric</th>
<th align="right">Value</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Number of Samples</td>
<td align="right">74</td>
</tr>
<tr class="even">
<td align="left">Number of Genes</td>
<td align="right">16302</td>
</tr>
<tr class="odd">
<td align="left">Average number of reads per cell</td>
<td align="right">16292</td>
</tr>
<tr class="even">
<td align="left">Average number of genes per cell</td>
<td align="right">7539</td>
</tr>
<tr class="odd">
<td align="left">Samples with &lt;1700 detected genes</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">Genes with no expression across all samples</td>
<td align="right">157</td>
</tr>
</tbody>
</table>
</div>
<div id="visualization" class="section level2">
<h2 class="hasAnchor">
<a href="#visualization" class="anchor"></a>Visualization</h2>
<p>Initially, there are no reduced dimensionality datasets stored in the object</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">reducedDims</span>(maits_subset)</code></pre></div>
<pre><code>## List of length 0</code></pre>
<p>PCA and t-SNE can be added to the object with the getPCA() and getTSNE() functions:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">maits_subset &lt;-<span class="st"> </span><span class="kw"><a href="../reference/getPCA.html">getPCA</a></span>(maits_subset, <span class="dt">useAssay =</span> <span class="st">"logtpm"</span>,
                       <span class="dt">reducedDimName =</span> <span class="st">"PCA_logtpm"</span>)
maits_subset &lt;-<span class="st"> </span><span class="kw"><a href="../reference/getPCA.html">getTSNE</a></span>(maits_subset, <span class="dt">useAssay =</span> <span class="st">"logtpm"</span>,
                        <span class="dt">reducedDimName =</span> <span class="st">"TSNE_logtpm"</span>)
<span class="kw">reducedDims</span>(maits_subset)</code></pre></div>
<pre><code>## List of length 2
## names(2): PCA_logtpm TSNE_logtpm</code></pre>
<div id="pca" class="section level3">
<h3 class="hasAnchor">
<a href="#pca" class="anchor"></a>PCA</h3>
<p>PCA data can be visualized with the plotPCA() function:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/getPCA.html">plotPCA</a></span>(maits_subset, <span class="dt">reducedDimName =</span> <span class="st">"PCA_logtpm"</span>, <span class="dt">colorBy =</span> <span class="st">"condition"</span>)</code></pre></div>
<p><img src="v02-Processing_and_Visualizing_Data_in_the_SingleCellTK_files/figure-html/maits_pca-1.png" width="700"></p>
</div>
<div id="t-sne" class="section level3">
<h3 class="hasAnchor">
<a href="#t-sne" class="anchor"></a>t-SNE</h3>
<p>t-SNE data can be visualized with the plotTSNE() function:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/getPCA.html">plotTSNE</a></span>(maits_subset, <span class="dt">reducedDimName =</span> <span class="st">"TSNE_logtpm"</span>, <span class="dt">colorBy =</span> <span class="st">"condition"</span>)</code></pre></div>
<p><img src="v02-Processing_and_Visualizing_Data_in_the_SingleCellTK_files/figure-html/maits_tsne-1.png" width="700"></p>
</div>
</div>
<div id="converting-gene-names" class="section level2">
<h2 class="hasAnchor">
<a href="#converting-gene-names" class="anchor"></a>Converting Gene Names</h2>
<p>The singleCellTK has the ability to convert gene ids to various formats using the org.*.eg.db Bioconductor annotation packages. These packages are not installed by default, so these must be manually installed before this function will work.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">suppressPackageStartupMessages</span>({
  <span class="kw">library</span>(org.Hs.eg.db)
})
maits_entrez &lt;-<span class="st"> </span>maits_subset
maits_subset &lt;-<span class="st"> </span><span class="kw"><a href="../reference/convertGeneIDs.html">convertGeneIDs</a></span>(maits_subset, <span class="dt">inSymbol =</span> <span class="st">"ENTREZID"</span>,
                               <span class="dt">outSymbol =</span> <span class="st">"SYMBOL"</span>, <span class="dt">database =</span> <span class="st">"org.Hs.eg.db"</span>)
<span class="co">#to remove confusion for MAST about the gene name:</span>
<span class="kw">rowData</span>(maits_subset)<span class="op">$</span>primerid &lt;-<span class="st"> </span><span class="ot">NULL</span></code></pre></div>
</div>
<div id="differential-expression-with-mast" class="section level2">
<h2 class="hasAnchor">
<a href="#differential-expression-with-mast" class="anchor"></a>Differential Expression with MAST</h2>
<p>MAST is a popular package for performing differential expression analysis on scRNA-Seq data that models the effect of dropouts using a bimodal distribution and by including the cellular detection rate into the differential expression model. Functions in the toolkit allow you to perform this analysis on a SCtkExperiemnt object.</p>
<div id="adaptive-thresholding" class="section level3">
<h3 class="hasAnchor">
<a href="#adaptive-thresholding" class="anchor"></a>Adaptive Thresholding</h3>
<p>First, an adaptive threshold is calculated by binning genes with similar expression levels.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">thresholds &lt;-<span class="st"> </span><span class="kw"><a href="../reference/MAST.html">thresholdGenes</a></span>(maits_subset, <span class="dt">useAssay =</span> <span class="st">"logtpm"</span>)
<span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">5</span>, <span class="dv">4</span>))
<span class="kw">plot</span>(thresholds)
<span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">1</span>))</code></pre></div>
<p><img src="v02-Processing_and_Visualizing_Data_in_the_SingleCellTK_files/figure-html/maits_thresh-1.png" width="700"></p>
</div>
<div id="run-mast" class="section level3">
<h3 class="hasAnchor">
<a href="#run-mast" class="anchor"></a>Run MAST</h3>
<p>MAST analysis can be run with a single function</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">mast_results &lt;-<span class="st"> </span><span class="kw"><a href="../reference/MAST.html">MAST</a></span>(maits_subset, <span class="dt">condition =</span> <span class="st">"condition"</span>, <span class="dt">useThresh =</span> <span class="ot">TRUE</span>,
                     <span class="dt">useAssay =</span> <span class="st">"logtpm"</span>)</code></pre></div>
<p>The resulting significantly differentially expressed genes can be visualized using a violin plot, linear model, or heatmap:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/MAST.html">MASTviolin</a></span>(maits_subset, <span class="dt">useAssay =</span> <span class="st">"logtpm"</span>, <span class="dt">fcHurdleSig =</span> mast_results,
           <span class="dt">threshP =</span> <span class="ot">TRUE</span>, <span class="dt">condition =</span> <span class="st">"condition"</span>)</code></pre></div>
<p><img src="v02-Processing_and_Visualizing_Data_in_the_SingleCellTK_files/figure-html/maits_violin-1.png" width="700"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/MAST.html">MASTregression</a></span>(maits_subset, <span class="dt">useAssay =</span> <span class="st">"logtpm"</span>, <span class="dt">fcHurdleSig =</span> mast_results,
               <span class="dt">threshP =</span> <span class="ot">TRUE</span>, <span class="dt">condition =</span> <span class="st">"condition"</span>)</code></pre></div>
<p><img src="v02-Processing_and_Visualizing_Data_in_the_SingleCellTK_files/figure-html/maits_lm-1.png" width="700"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/plotDiffEx.html">plotDiffEx</a></span>(maits_subset, <span class="dt">useAssay =</span> <span class="st">"logtpm"</span>, <span class="dt">condition =</span> <span class="st">"condition"</span>,
           <span class="dt">geneList =</span> mast_results<span class="op">$</span>Gene[<span class="dv">1</span><span class="op">:</span><span class="dv">100</span>], <span class="dt">annotationColors =</span> <span class="st">"auto"</span>,
           <span class="dt">displayRowLabels =</span> <span class="ot">FALSE</span>, <span class="dt">displayColumnLabels =</span> <span class="ot">FALSE</span>)</code></pre></div>
<p><img src="v02-Processing_and_Visualizing_Data_in_the_SingleCellTK_files/figure-html/maits_heatmap-1.png" width="700"></p>
<p>Among the top differentially expressed genes was interferon gamma, a cytokine that is known to be produced in response to stimulation.</p>
</div>
<div id="pathway-activity-with-gsva" class="section level3">
<h3 class="hasAnchor">
<a href="#pathway-activity-with-gsva" class="anchor"></a>Pathway Activity with GSVA</h3>
<p>The singleCellTK supports pathway activity analysis using the GSVA package. Currently, the toolkit supports performing this analysis on human datasets with entrez IDs. Data can be visualized as a violin plot or a heatmap.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">gsvaRes &lt;-<span class="st"> </span><span class="kw"><a href="../reference/gsvaSCE.html">gsvaSCE</a></span>(maits_entrez, <span class="dt">useAssay =</span> <span class="st">"logtpm"</span>,
                   <span class="st">"MSigDB c2 (Human, Entrez ID only)"</span>,
                   <span class="kw">c</span>(<span class="st">"KEGG_PROTEASOME"</span>,
                     <span class="st">"REACTOME_VIF_MEDIATED_DEGRADATION_OF_APOBEC3G"</span>,
                     <span class="st">"REACTOME_P53_INDEPENDENT_DNA_DAMAGE_RESPONSE"</span>,
                     <span class="st">"BIOCARTA_PROTEASOME_PATHWAY"</span>,
                     <span class="st">"REACTOME_METABOLISM_OF_AMINO_ACIDS"</span>,
                     <span class="st">"REACTOME_REGULATION_OF_ORNITHINE_DECARBOXYLASE"</span>,
                     <span class="st">"REACTOME_CYTOSOLIC_TRNA_AMINOACYLATION"</span>,
                     <span class="st">"REACTOME_STABILIZATION_OF_P53"</span>,
                     <span class="st">"REACTOME_SCF_BETA_TRCP_MEDIATED_DEGRADATION_OF_EMI1"</span>),
                    <span class="dt">parallel.sz=</span><span class="dv">1</span>)</code></pre></div>
<pre><code>## Warning in .local(expr, gset.idx.list, ...): 157 genes with constant
## expression values throuhgout the samples.</code></pre>
<pre><code>## Warning in .local(expr, gset.idx.list, ...): Since argument method!
## ="ssgsea", genes with constant expression values are discarded.</code></pre>
<pre><code>## Estimating GSVA scores for 9 gene sets.
## Computing observed enrichment scores
## Estimating ECDFs with Gaussian kernels
## Using parallel with 1 cores
## 
  |                                                                       
  |                                                                 |   0%
  |                                                                       
  |=======                                                          |  11%
  |                                                                       
  |==============                                                   |  22%
  |                                                                       
  |======================                                           |  33%
  |                                                                       
  |=============================                                    |  44%
  |                                                                       
  |====================================                             |  56%
  |                                                                       
  |===========================================                      |  67%
  |                                                                       
  |===================================================              |  78%
  |                                                                       
  |==========================================================       |  89%
  |                                                                       
  |=================================================================| 100%</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">set.seed</span>(<span class="dv">1234</span>)
<span class="kw"><a href="../reference/gsvaSCE.html">gsvaPlot</a></span>(maits_subset, gsvaRes, <span class="st">"Violin"</span>, <span class="st">"condition"</span>)</code></pre></div>
<p><img src="v02-Processing_and_Visualizing_Data_in_the_SingleCellTK_files/figure-html/maits_gsva-1.png" width="700"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/gsvaSCE.html">gsvaPlot</a></span>(maits_subset, gsvaRes, <span class="st">"Heatmap"</span>, <span class="st">"condition"</span>)</code></pre></div>
<p><img src="v02-Processing_and_Visualizing_Data_in_the_SingleCellTK_files/figure-html/maits_gsva-2.png" width="700"></p>
<p>Among the top pathways that showed increased activity in the stimulated cells was KEGG_PROTEASOME, indicating proteasome related genes showed increased activity in the stimulated T cells. This pathway includes interferon gamma.</p>
</div>
</div>
</div>
<div id="batch-effects-example" class="section level1">
<h1 class="hasAnchor">
<a href="#batch-effects-example" class="anchor"></a>Batch Effects Example</h1>
<p>It is possible to use ComBat within the Single Cell Toolkit. This support is experimental, since ComBat was not designed for scRNA-Seq. Here, we will load the bladderbatch example data into a SingleCellExperiment object.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(bladderbatch)
<span class="kw">data</span>(bladderdata)
dat &lt;-<span class="st"> </span>bladderEset

pheno &lt;-<span class="st"> </span><span class="kw">pData</span>(dat)
edata &lt;-<span class="st"> </span><span class="kw">exprs</span>(dat)
bladder_sctke &lt;-<span class="st"> </span><span class="kw"><a href="../reference/createSCE.html">createSCE</a></span>(<span class="dt">assayFile =</span> edata,
                           <span class="dt">annotFile =</span> pheno,
                           <span class="dt">assayName =</span> <span class="st">"microarray"</span>,
                           <span class="dt">inputDataFrames =</span> <span class="ot">TRUE</span>,
                           <span class="dt">createLogCounts =</span> <span class="ot">FALSE</span>)</code></pre></div>
<p>The plotBatchVariance() function can be used to plot the percent variation explained by condition and batch across the dataset.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/plotBatchVariance.html">plotBatchVariance</a></span>(bladder_sctke, <span class="dt">useAssay=</span><span class="st">"microarray"</span>,
                  <span class="dt">batch=</span><span class="st">"batch"</span>, <span class="dt">condition =</span> <span class="st">"cancer"</span>)</code></pre></div>
<p><img src="v02-Processing_and_Visualizing_Data_in_the_SingleCellTK_files/figure-html/plot_var_microarray-1.png" width="700"></p>
<p>The ComBatSCE() function can then be used to correct for batch effects</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">assay</span>(bladder_sctke, <span class="st">"combat"</span>) &lt;-<span class="st"> </span><span class="kw"><a href="../reference/ComBatSCE.html">ComBatSCE</a></span>(<span class="dt">inSCE =</span> bladder_sctke,
                                            <span class="dt">batch =</span> <span class="st">"batch"</span>,
                                            <span class="dt">useAssay =</span> <span class="st">"microarray"</span>,
                                            <span class="dt">covariates =</span> <span class="st">"cancer"</span>)</code></pre></div>
<pre><code>## Standardizing Data across genes</code></pre>
<p>After batch correction, a larger percentage of the explained variation can be explained by the condition</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/plotBatchVariance.html">plotBatchVariance</a></span>(bladder_sctke, <span class="dt">useAssay=</span><span class="st">"combat"</span>,
                  <span class="dt">batch=</span><span class="st">"batch"</span>, <span class="dt">condition =</span> <span class="st">"cancer"</span>)</code></pre></div>
<p><img src="v02-Processing_and_Visualizing_Data_in_the_SingleCellTK_files/figure-html/plot_var_postcombat-1.png" width="700"></p>
</div>
<div id="session-info" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session info</h1>
<pre><code>## R version 3.5.1 (2018-07-02)
## Platform: x86_64-apple-darwin15.6.0 (64-bit)
## Running under: macOS High Sierra 10.13.6
## 
## Matrix products: default
## BLAS: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRblas.0.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] parallel  stats4    stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] bladderbatch_1.18.0         GSEABase_1.42.0            
##  [3] graph_1.58.0                annotate_1.58.0            
##  [5] XML_3.98-1.16               org.Hs.eg.db_3.6.0         
##  [7] AnnotationDbi_1.42.1        xtable_1.8-2               
##  [9] singleCellTK_1.1.18         MAST_1.6.1                 
## [11] SingleCellExperiment_1.2.0  SummarizedExperiment_1.10.1
## [13] DelayedArray_0.6.5          BiocParallel_1.14.2        
## [15] matrixStats_0.54.0          Biobase_2.40.0             
## [17] GenomicRanges_1.32.6        GenomeInfoDb_1.16.0        
## [19] IRanges_2.14.10             S4Vectors_0.18.3           
## [21] BiocGenerics_0.26.0         BiocStyle_2.8.2            
## 
## loaded via a namespace (and not attached):
##  [1] nlme_3.1-137           bitops_1.0-6           fs_1.2.5              
##  [4] GSVA_1.28.0            bit64_0.9-7            RColorBrewer_1.1-2    
##  [7] rprojroot_1.3-2        tools_3.5.1            backports_1.1.2       
## [10] R6_2.2.2               mgcv_1.8-24            DBI_1.0.0             
## [13] lazyeval_0.2.1         colorspace_1.3-2       GetoptLong_0.1.7      
## [16] tidyselect_0.2.4       bit_1.1-14             compiler_3.5.1        
## [19] xml2_1.2.0             desc_1.2.0             labeling_0.3          
## [22] bookdown_0.7           scales_1.0.0           genefilter_1.62.0     
## [25] pkgdown_1.1.0          commonmark_1.5         stringr_1.3.1         
## [28] digest_0.6.15          rmarkdown_1.10         XVector_0.20.0        
## [31] pkgconfig_2.0.2        htmltools_0.3.6        limma_3.36.2          
## [34] highr_0.7              rlang_0.2.2            GlobalOptions_0.1.0   
## [37] rstudioapi_0.7         RSQLite_2.1.1          shiny_1.1.0           
## [40] shape_1.4.4            bindr_0.1.1            dplyr_0.7.6           
## [43] RCurl_1.95-4.11        magrittr_1.5           GenomeInfoDbData_1.1.0
## [46] Matrix_1.2-14          Rcpp_0.12.18           munsell_0.5.0         
## [49] abind_1.4-5            stringi_1.2.4          yaml_2.2.0            
## [52] MASS_7.3-50            zlibbioc_1.26.0        Rtsne_0.13            
## [55] plyr_1.8.4             grid_3.5.1             blob_1.1.1            
## [58] promises_1.0.1         crayon_1.3.4           lattice_0.20-35       
## [61] splines_3.5.1          circlize_0.4.4         knitr_1.20            
## [64] ComplexHeatmap_1.18.1  pillar_1.3.0           rjson_0.2.20          
## [67] geneplotter_1.58.0     reshape2_1.4.3         glue_1.3.0            
## [70] evaluate_0.11          data.table_1.11.4      httpuv_1.4.5          
## [73] gtable_0.2.0           purrr_0.2.5            assertthat_0.2.0      
## [76] ggplot2_3.0.0          xfun_0.3               mime_0.5              
## [79] GSVAdata_1.16.0        later_0.7.3            roxygen2_6.1.0        
## [82] survival_2.42-6        tibble_1.4.2           shinythemes_1.1.1     
## [85] memoise_1.1.0          bindrcpp_0.2.2         sva_3.28.0</code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#introduction">Introduction</a></li>
      <li>
<a href="#maits-example">MAITS Example</a><ul class="nav nav-pills nav-stacked">
<li><a href="#summarizetable">summarizeTable</a></li>
      <li><a href="#filtering-by-annotation">Filtering by Annotation</a></li>
      <li><a href="#visualization">Visualization</a></li>
      <li><a href="#converting-gene-names">Converting Gene Names</a></li>
      <li><a href="#differential-expression-with-mast">Differential Expression with MAST</a></li>
      </ul>
</li>
      <li><a href="#batch-effects-example">Batch Effects Example</a></li>
      <li><a href="#session-info">Session info</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by David Jenkins, Tyler Faits, Mohammed Muzamil Khan, Emma Briars, Sebastian Carrasco Pro, W. Evan Johnson.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  

  </body>
</html>
