<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Data Summary and Filtering Tab • singleCellTK</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootswatch/3.3.7/yeti/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Data Summary and Filtering Tab">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">singleCellTK</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">1.1.18</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/v01-Introduction_to_singleCellTK.html">Introduction to singleCellTK</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Interactive Shiny Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/v03-tab01_Upload.html">1: Upload</a>
    </li>
    <li>
      <a href="../articles/v04-tab02_Data-Summary-and-Filtering.html">2: Data Summary and Filtering</a>
    </li>
    <li>
      <a href="../articles/v05-tab03_Visualization-and-Clustering.html">3: Visualization and Clustering</a>
    </li>
    <li>
      <a href="../articles/v06-tab04_Batch-Correction.html">4: Batch Correction</a>
    </li>
    <li>
      <a href="../articles/v07-tab05_Differential-Expression.html">5: Differential Expression</a>
    </li>
    <li>
      <a href="../articles/v08-tab06_Pathway-Activity-Analysis.html">6: Pathway Activity Analysis</a>
    </li>
    <li>
      <a href="../articles/v09-tab07_Sample-Size.html">7: Sample Size</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Command Line Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/v02-Processing_and_Visualizing_Data_in_the_SingleCellTK.html">Processing and Visualizing Data</a>
    </li>
    <li>
      <a href="../articles/v10-Aligning_and_Quantifying_scRNA-Seq_Data.html">Aligning and Quantifying scRNA-Seq Data</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/compbiomed/singleCellTK">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Data Summary and Filtering Tab</h1>
            <h3 class="subtitle">Single Cell Toolkit</h3>
                        <h4 class="author">David Jenkins</h4>
            <address class="author_afil">
      The Section of Computational Biomedicine, Boston University School of Medicine, Boston, MA<br><a class="author_email" href="mailto:#"></a><a href="mailto:dfj@bu.edu">dfj@bu.edu</a>
      </address>
                              <h4 class="author">Tyler Faits</h4>
            <address class="author_afil">
      The Section of Computational Biomedicine, Boston University School of Medicine, Boston, MA<br><h4 class="author">W. Evan Johnson</h4>
            <address class="author_afil">
      The Section of Computational Biomedicine, Boston University School of Medicine, Boston, MA<br><div class="hidden name"><code>v04-tab02_Data-Summary-and-Filtering.Rmd</code></div>

    </address>
</address>
</div>

    
    
<div class="figure">
<img src="https://imgur.com/611yD5H.png">
</div>
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>The data summary and filtering tab provides several summary statistics and ways to filter your data. The summary statistics and filtering are performed on the selected assay, which can be changed using the “Select Assay” drop-down.</p>
</div>
<div id="data-summary-table" class="section level1">
<h1 class="hasAnchor">
<a href="#data-summary-table" class="anchor"></a>Data summary table</h1>
<p><img src="https://imgur.com/VBT0gKu.png" width="400"></p>
<p>The summary table provides several summary statistics about your data including:</p>
<ul>
<li>The number of samples</li>
<li>The number of genes</li>
<li>The average number of reads per cell</li>
<li>The average number of genes per cell</li>
<li>Samples with fewer than the selected cutoff expressed genes</li>
<li>Genes with no expression across all samples</li>
</ul>
</div>
<div id="filtering" class="section level1">
<h1 class="hasAnchor">
<a href="#filtering" class="anchor"></a>Filtering</h1>
<p>Several filtering options are available</p>
<div id="cutoffs" class="section level2">
<h2 class="hasAnchor">
<a href="#cutoffs" class="anchor"></a>Cutoffs</h2>
<p><img src="https://imgur.com/0VWS0qP.png" width="400"></p>
<p>Data can be filtered by user selected cutoff values including:</p>
<ul>
<li>Remove genes with no expression across all samples. These genes are uninformative</li>
<li>Minimum detected genes per sample. Samples with few expressed genes could indicate an empty well.</li>
<li>Percent low gene expression. This filter will remove the genes with the least expression in the dataset</li>
</ul>
<p>When you have set the cutoff filters to the desired level, you can filter data with the “Filter Data” button.</p>
</div>
<div id="delete-outliers" class="section level2">
<h2 class="hasAnchor">
<a href="#delete-outliers" class="anchor"></a>Delete Outliers</h2>
<p>If there are specific cells that should be excluded from the dataset, you can select them and they will be removed when “Filter Data” is clicked.</p>
</div>
<div id="reset" class="section level2">
<h2 class="hasAnchor">
<a href="#reset" class="anchor"></a>Reset</h2>
<p>The SCTK saves a copy of the originally uploaded data, which can be restored by clicking the “Reset” button.</p>
</div>
<div id="filter-samples-by-annotation" class="section level2">
<h2 class="hasAnchor">
<a href="#filter-samples-by-annotation" class="anchor"></a>Filter samples by annotation</h2>
<p><img src="https://imgur.com/4I8V6Um.png" width="400"></p>
<p>Data stored in the annotation data frame can be used for sample filtering. Choose a column from the annotation data frame, the values from this column to keep, and click “Filter.”</p>
</div>
<div id="filter-genes-by-feature-annotation" class="section level2">
<h2 class="hasAnchor">
<a href="#filter-genes-by-feature-annotation" class="anchor"></a>Filter genes by feature annotation</h2>
<p><img src="https://imgur.com/zUew8tj.png" width="400"></p>
<p>Similarly, gene annotations can be used to filter genes. Select a column from the row data frame and the values to keep and click “Filter.”</p>
</div>
<div id="convert-gene-annotations" class="section level2">
<h2 class="hasAnchor">
<a href="#convert-gene-annotations" class="anchor"></a>Convert gene annotations</h2>
<p><img src="https://imgur.com/ukBNdLp.png" width="400"></p>
<p>The SCTK can use annotation data from Bioconductor annotation packages such as <a href="https://www.bioconductor.org/packages/release/data/annotation/html/org.Hs.eg.db.html">org.Hs.eg.db</a> to convert gene annotations between standard annotation types such as Entrez gene ids, Ensembl gene ids, or gene symbols. By default, these packages are not installed, so you will need to manually install the package for your appropriate organism. Instructions for installation can be found on the <a href="https://www.bioconductor.org/install/#install-bioconductor-packages">Bioconductor website</a></p>
</div>
<div id="delete-an-annotation-column" class="section level2">
<h2 class="hasAnchor">
<a href="#delete-an-annotation-column" class="anchor"></a>Delete an annotation column</h2>
<p><img src="https://imgur.com/EXr0XMu.png" width="400"></p>
<p>You can remove an unnecessary or unwanted annotation column by selecting it and clicking the “Delete Column” button.</p>
</div>
<div id="randomly-subset" class="section level2">
<h2 class="hasAnchor">
<a href="#randomly-subset" class="anchor"></a>Randomly Subset</h2>
<p><img src="https://i.imgur.com/OFgutav.png" width="400"></p>
<p>If you are working with a large dataset, you may want to randomly subset the data for exploratory analysis. Specify the number of samples that you wish to keep in the subset data frame and click the “Subset Data” button.</p>
</div>
<div id="download-sctkexperiment" class="section level2">
<h2 class="hasAnchor">
<a href="#download-sctkexperiment" class="anchor"></a>Download SCtkExperiment</h2>
<p><img src="https://imgur.com/WvlaNN3.png" width="400"></p>
<p>To export the SCtkExperiment object from the SCTK, you can download the object in RDS format by clicking the “Download SCtkExperiment” button. This file can be loaded into the R console for downstream analysis.</p>
</div>
</div>
<div id="assay-details" class="section level1">
<h1 class="hasAnchor">
<a href="#assay-details" class="anchor"></a>Assay Details</h1>
<p>The assay details tab describes the available assays and reduced dimensionality data and allows the user to modify the SCTK experiment object</p>
<div id="available-assays-and-reduceddim-data" class="section level2">
<h2 class="hasAnchor">
<a href="#available-assays-and-reduceddim-data" class="anchor"></a>Available Assays and reducedDim Data</h2>
<p><img src="https://imgur.com/XnmgJIJ.png" width="400"></p>
<p>The assays and reduced dimension data listed on this tab are stored in the underlying SCTK experiment object and are available on the tabs in the app.</p>
</div>
<div id="add-and-delete-assays-and-reduceddim-data" class="section level2">
<h2 class="hasAnchor">
<a href="#add-and-delete-assays-and-reduceddim-data" class="anchor"></a>Add and Delete Assays and reducedDim Data</h2>
<p><img src="https://imgur.com/39V7UrX.png" width="200"></p>
<p>If a count matrix is available, users can add a log(counts), CPM (counts per million), and log(CPM) matrix directly through the app. Users can also delete unwanted assays or reducedDim objects on this tab</p>
</div>
</div>
<div id="annotation-data" class="section level1">
<h1 class="hasAnchor">
<a href="#annotation-data" class="anchor"></a>Annotation Data</h1>
<p>The annotation tab displays the annotation matrix.</p>
<p><img src="https://imgur.com/prYLR5D.png" width="600"></p>
<div id="convert-annotation-data-columns" class="section level2">
<h2 class="hasAnchor">
<a href="#convert-annotation-data-columns" class="anchor"></a>Convert Annotation Data Columns</h2>
<p>For plotting, annotations that contain character values are interpreted as factors and will be displayed with discrete colors. Numeric annotations will be plotted with a continuous color bar. To change a numeric value to a factor, select the annotation column and change the “Field Type” from numeric to factor.</p>
</div>
<div id="replace-annotation-data" class="section level2">
<h2 class="hasAnchor">
<a href="#replace-annotation-data" class="anchor"></a>Replace Annotation Data</h2>
<p>To modify the annotation data frame, download the annotation data using the “Download Annotation Data” button, modify the data using a text editor or Excel, save the data as a .csv file, and re-upload the data in the “Upload and replace the annotation data” field.</p>
</div>
</div>
<div id="session-info" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session info</h1>
<pre><code>## R version 3.5.1 (2018-07-02)
## Platform: x86_64-apple-darwin15.6.0 (64-bit)
## Running under: macOS High Sierra 10.13.6
## 
## Matrix products: default
## BLAS: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRblas.0.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
## [1] BiocStyle_2.8.2
## 
## loaded via a namespace (and not attached):
##  [1] Rcpp_0.12.18     rstudioapi_0.7   knitr_1.20       xml2_1.2.0      
##  [5] magrittr_1.5     roxygen2_6.1.0   MASS_7.3-50      R6_2.2.2        
##  [9] rlang_0.2.2      stringr_1.3.1    tools_3.5.1      xfun_0.3        
## [13] htmltools_0.3.6  commonmark_1.5   yaml_2.2.0       rprojroot_1.3-2 
## [17] digest_0.6.15    assertthat_0.2.0 bookdown_0.7     pkgdown_1.1.0   
## [21] crayon_1.3.4     fs_1.2.5         memoise_1.1.0    evaluate_0.11   
## [25] rmarkdown_1.10   stringi_1.2.4    compiler_3.5.1   desc_1.2.0      
## [29] backports_1.1.2</code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#introduction">Introduction</a></li>
      <li><a href="#data-summary-table">Data summary table</a></li>
      <li>
<a href="#filtering">Filtering</a><ul class="nav nav-pills nav-stacked">
<li><a href="#cutoffs">Cutoffs</a></li>
      <li><a href="#delete-outliers">Delete Outliers</a></li>
      <li><a href="#reset">Reset</a></li>
      <li><a href="#filter-samples-by-annotation">Filter samples by annotation</a></li>
      <li><a href="#filter-genes-by-feature-annotation">Filter genes by feature annotation</a></li>
      <li><a href="#convert-gene-annotations">Convert gene annotations</a></li>
      <li><a href="#delete-an-annotation-column">Delete an annotation column</a></li>
      <li><a href="#randomly-subset">Randomly Subset</a></li>
      <li><a href="#download-sctkexperiment">Download SCtkExperiment</a></li>
      </ul>
</li>
      <li>
<a href="#assay-details">Assay Details</a><ul class="nav nav-pills nav-stacked">
<li><a href="#available-assays-and-reduceddim-data">Available Assays and reducedDim Data</a></li>
      <li><a href="#add-and-delete-assays-and-reduceddim-data">Add and Delete Assays and reducedDim Data</a></li>
      </ul>
</li>
      <li>
<a href="#annotation-data">Annotation Data</a><ul class="nav nav-pills nav-stacked">
<li><a href="#convert-annotation-data-columns">Convert Annotation Data Columns</a></li>
      <li><a href="#replace-annotation-data">Replace Annotation Data</a></li>
      </ul>
</li>
      <li><a href="#session-info">Session info</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by David Jenkins, Tyler Faits, Mohammed Muzamil Khan, Emma Briars, Sebastian Carrasco Pro, W. Evan Johnson.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  

  </body>
</html>
