<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Differential Expression Tab • singleCellTK</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootswatch/3.3.7/yeti/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../jquery.sticky-kit.min.js"></script><script src="../pkgdown.js"></script><meta property="og:title" content="Differential Expression Tab">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">singleCellTK</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">0.99.10</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/v01-Introduction_to_singleCellTK.html">Introduction to singleCellTK</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Interactive Shiny Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/v03-tab01_Upload.html">1: Upload</a>
    </li>
    <li>
      <a href="../articles/v04-tab02_Data-Summary-and-Filtering.html">2: Data Summary and Filtering</a>
    </li>
    <li>
      <a href="../articles/v05-tab03_Dimensionality-Reduction-and-Clustering.html">3: Dimensionality Reduction and Clustering</a>
    </li>
    <li>
      <a href="../articles/v06-tab04_Batch-Correction.html">4: Batch Correction</a>
    </li>
    <li>
      <a href="../articles/v07-tab05_Differential-Expression.html">5: Differential Expression</a>
    </li>
    <li>
      <a href="../articles/v08-tab06_Pathway-Activity-Analysis.html">6: Pathway Activity Analysis</a>
    </li>
    <li>
      <a href="../articles/v09-tab07_Sample-Size.html">7: Sample Size</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Command Line Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/v02-Processing_and_Visualizing_Data_in_the_SingleCellTK.html">Processing and Visualizing Data</a>
    </li>
    <li>
      <a href="../articles/v10-Aligning_and_Quantifying_scRNA-Seq_Data.html">Aligning and Quantifying scRNA-Seq Data</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/compbiomed/singleCellTK">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Differential Expression Tab</h1>
            <h3 class="subtitle">Single Cell Toolkit</h3>
                        <h4 class="author">David Jenkins</h4>
            <address class="author_afil">
      The Section of Computational Biomedicine, Boston University School of Medicine, Boston, MA<br><a class="author_email" href="mailto:#"></a><a href="mailto:dfj@bu.edu">dfj@bu.edu</a>
      </address>
                              <h4 class="author">Tyler Faits</h4>
            <address class="author_afil">
      The Section of Computational Biomedicine, Boston University School of Medicine, Boston, MA<br><h4 class="author">W. Evan Johnson</h4>
            <address class="author_afil">
      The Section of Computational Biomedicine, Boston University School of Medicine, Boston, MA<br>
</address>
</address>
</div>

    
    
<div class="figure">
<img src="https://imgur.com/F0wG77W.png">
</div>
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>Differential expression analysis can identify genes that are significantly up or down regulated between conditions. While many differential expression algorithms exist, their performance may vary on scRNA-Seq datasets.</p>
</div>
<div id="limma-deseq2-and-anova" class="section level1">
<h1 class="hasAnchor">
<a href="#limma-deseq2-and-anova" class="anchor"></a>limma, DESeq2, and ANOVA</h1>
<p>Users can apply common differential expression algorithms <a href="https://bioconductor.org/packages/release/bioc/html/limma.html">limma</a>, <a href="https://bioconductor.org/packages/release/bioc/html/DESeq2.html">DESeq2</a>, or perform an ANOVA to identify differentially expressed genes by selecting one or multiple condition variables present in the annotation information. After choosing the assay of interest using the “Select Assay” field and the algorithm in the “Select Method” field, choose the experimental condition and any additional covariates to use in the differential expression model. For limma analysis with a condition with more than two categories, a “Factor of Interest” must be selected. For DESeq2 analysis with more than two categories, users can select one of three modes. In Biomarker mode, select a factor of interest. The resulting genes will be differentially expressed between the factor of interest and all other cells. For factor of interest vs control factor analysis, the user selects a factor of interest and a control factor and the resulting genes will be significantly differentially expressed between these two-factor levels. Finally, for ANOVA, DESeq2 will be run using a likelihood ratio test. Users can customize the differential expression results by changing the number of genes to return, the p-value significance cutoff, and the p-value correction method applied to the results.</p>
<div id="visualization" class="section level2">
<h2 class="hasAnchor">
<a href="#visualization" class="anchor"></a>Visualization</h2>
<div class="figure">
<img src="https://imgur.com/v9hEOv1.png">
</div>
<p>The resulting gene list is displayed as a table in the “Results Table” tab and also in a heatmap in the “Heatmap” tab, which can also be customized using the options available in the “Options” tab. Users can download the gene list directly or create a biomarker list for a specific cell type or cell cluster, which can be stored in the gene annotation information in the SCtkExperiment object.</p>
</div>
</div>
<div id="mast" class="section level1">
<h1 class="hasAnchor">
<a href="#mast" class="anchor"></a>MAST</h1>
<div class="figure">
<img src="https://imgur.com/78hGZ96.png">
</div>
<p><a href="https://www.bioconductor.org/packages/release/bioc/html/MAST.html">MAST</a>, Model-based Analysis of Single-cell Transcriptomics, is a differential expression analysis tool specifically designed for single cell RNA-Seq data, which uses a hurdle model to account for the missingness in scRNA-Seq data. MAST has been implemented within the SCTK. Users can choose whether to use MAST’s adaptive thresholding model, choose fold change and expression thresholds, and identify significant genes based on conditions present in the annotation information provided. The results are presented in a table, violin plots, or visualized in a heatmap. For detailed information about MAST analysis, see the <a href="https://www.bioconductor.org/packages/release/bioc/html/MAST.html">MAST documentation</a>.</p>
<div class="figure">
<img src="https://imgur.com/MjRl0MN.png">
</div>
<div class="figure">
<img src="https://imgur.com/VhfMnzS.png">
</div>
</div>
<div id="saving-a-biomarker" class="section level1">
<h1 class="hasAnchor">
<a href="#saving-a-biomarker" class="anchor"></a>Saving a Biomarker</h1>
<p><img src="https://imgur.com/ErIRBDH.png" width="350"></p>
<p>After differential expression analysis has been performed, the resulting gene list can be downloaded or stored in the SCtkExperiment object in the gene annotation data frame.</p>
</div>
<div id="session-info" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session info</h1>
<pre><code>## R Under development (unstable) (2018-03-09 r74376)
## Platform: x86_64-apple-darwin15.6.0 (64-bit)
## Running under: macOS High Sierra 10.13.4
## 
## Matrix products: default
## BLAS: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRblas.0.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
## [1] BiocStyle_2.7.8
## 
## loaded via a namespace (and not attached):
##  [1] Rcpp_0.12.16       rstudioapi_0.7     knitr_1.20        
##  [4] xml2_1.2.0         magrittr_1.5       roxygen2_6.0.1    
##  [7] MASS_7.3-49        R6_2.2.2           rlang_0.2.0       
## [10] stringr_1.3.0      tools_3.5.0        xfun_0.1          
## [13] htmltools_0.3.6    commonmark_1.4     yaml_2.1.18       
## [16] rprojroot_1.3-2    digest_0.6.15      assertthat_0.2.0  
## [19] bookdown_0.7       pkgdown_0.1.0.9000 crayon_1.3.4      
## [22] fs_1.2.2           memoise_1.1.0      evaluate_0.10.1   
## [25] rmarkdown_1.9      stringi_1.1.7      compiler_3.5.0    
## [28] desc_1.1.1         backports_1.1.2</code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#introduction">Introduction</a></li>
      <li>
<a href="#limma-deseq2-and-anova">limma, DESeq2, and ANOVA</a><ul class="nav nav-pills nav-stacked">
<li><a href="#visualization">Visualization</a></li>
      </ul>
</li>
      <li><a href="#mast">MAST</a></li>
      <li><a href="#saving-a-biomarker">Saving a Biomarker</a></li>
      <li><a href="#session-info">Session info</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by David Jenkins, Tyler Faits, Emma Briars, Sebastian Carrasco Pro, W. Evan Johnson.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  

  </body>
</html>
