<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Align Single Cell RNA-Seq Data and Create a SCtkExperiment Object — alignSingleCellData • singleCellTK</title>

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.3.7/yeti/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- sticky kit -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



<meta property="og:title" content="Align Single Cell RNA-Seq Data and Create a SCtkExperiment Object — alignSingleCellData" />

<meta property="og:description" content="Align Single Cell RNA-Seq Data and Create a SCtkExperiment Object" />
<meta name="twitter:card" content="summary" />



<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">singleCellTK</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.3.7</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../articles/v01-Introduction_to_singleCellTK.html">Introduction to singleCellTK</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Interactive Shiny Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/v03-tab01_Upload.html">1: Upload</a>
    </li>
    <li>
      <a href="../articles/v04-tab02_Data-Summary-and-Filtering.html">2: Data Summary and Filtering</a>
    </li>
    <li>
      <a href="../articles/v05-tab03_Visualization-and-Clustering.html">3: Visualization and Clustering</a>
    </li>
    <li>
      <a href="../articles/v06-tab04_Batch-Correction.html">4: Batch Correction</a>
    </li>
    <li>
      <a href="../articles/v07-tab05_Differential-Expression.html">5: Differential Expression</a>
    </li>
    <li>
      <a href="../articles/v08-tab06_Pathway-Activity-Analysis.html">6: Pathway Activity Analysis</a>
    </li>
    <li>
      <a href="../articles/v09-tab07_Sample-Size.html">7: Sample Size</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Command Line Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/v02-Processing_and_Visualizing_Data_in_the_SingleCellTK.html">Processing and Visualizing Data</a>
    </li>
    <li>
      <a href="../articles/v10-Aligning_and_Quantifying_scRNA-Seq_Data.html">Aligning and Quantifying scRNA-Seq Data</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/compbiomed/singleCellTK">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Align Single Cell RNA-Seq Data and Create a SCtkExperiment Object</h1>
    
    <div class="hidden name"><code>alignSingleCellData.Rd</code></div>
    </div>

    <div class="ref-description">
    
    <p>Align Single Cell RNA-Seq Data and Create a SCtkExperiment Object</p>
    
    </div>

    <pre class="usage"><span class='fu'>alignSingleCellData</span>(<span class='no'>inputfile1</span>, <span class='kw'>inputfile2</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='no'>indexPath</span>,
  <span class='no'>gtfAnnotation</span>, <span class='kw'>outputDir</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='kw'>sampleAnnotations</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>featureAnnotations</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='kw'>threads</span> <span class='kw'>=</span> <span class='fl'>1</span>, <span class='kw'>saveBam</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>saveCountFiles</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>, <span class='kw'>isPairedEnd</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>)</pre>
    
    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>inputfile1</th>
      <td><p>An input file or list of files. Files can be fastq,
fastq.gz, or bam, but must all be of the same type. Sample names will be the
full file name, without _1.fastq.gz, .fastq.gz, _1.fastq, .fastq or .bam
endings.</p></td>
    </tr>
    <tr>
      <th>inputfile2</th>
      <td><p>If fastq files are provided in input list, a list of
corresponding paired fastq files, if applicable.</p></td>
    </tr>
    <tr>
      <th>indexPath</th>
      <td><p>Path to the Rsubread genome index.</p></td>
    </tr>
    <tr>
      <th>gtfAnnotation</th>
      <td><p>Path to the GTF gene annotation to use. This must
correspond to the genome specified in indexPath.</p></td>
    </tr>
    <tr>
      <th>outputDir</th>
      <td><p>If saveBam or saveCountFiles is TRUE, specify a
directory in which to save the output files.</p></td>
    </tr>
    <tr>
      <th>sampleAnnotations</th>
      <td><p>A data.frame of sample annotations, with samples
as rows and annotations in columns. The sample names must be identical to
and in the same order as the list of files in inputfile1. Alignment
statistics will be added to the annotation data frame.</p></td>
    </tr>
    <tr>
      <th>featureAnnotations</th>
      <td><p>An optional data.frame of probe annotations, with
probes as rows and probe annotations in columns.</p></td>
    </tr>
    <tr>
      <th>threads</th>
      <td><p>Number of threads to use during alignment. The default is 1.</p></td>
    </tr>
    <tr>
      <th>saveBam</th>
      <td><p>If TRUE, bam alignment files will be saved in the outputDir.
The default is FALSE.</p></td>
    </tr>
    <tr>
      <th>saveCountFiles</th>
      <td><p>If TRUE, per sample gene count files will be saved in
the outputDir. The default is FALSE.</p></td>
    </tr>
    <tr>
      <th>isPairedEnd</th>
      <td><p>If input files are .bam, indicate whether the input bam
files are paired end.</p></td>
    </tr>
    </table>
    
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>Object to import into the shiny app.</p>
    

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><span class='co'># NOT RUN {</span>
<span class='no'>singlecellobject</span> <span class='kw'>&lt;-</span> <span class='fu'>alignSingleCellData</span>(
  <span class='kw'>inputfile1</span> <span class='kw'>=</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/c'>c</a></span>(<span class='st'>"/path/to/sample1_1.fastq.gz"</span>,
                 <span class='st'>"/path/to/sample2_1.fastq.gz"</span>),
  <span class='kw'>inputfile2</span> <span class='kw'>=</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/c'>c</a></span>(<span class='st'>"/path/to/sample1_2.fastq.gz"</span>,
                 <span class='st'>"/path/to/sample2_2.fastq.gz"</span>),
  <span class='kw'>indexPath</span> <span class='kw'>=</span> <span class='st'>"/path/to/genome/index"</span>,
  <span class='kw'>gtfAnnotation</span> <span class='kw'>=</span> <span class='st'>"/path/to/gene/annotations.gtf"</span>,
  <span class='kw'>sampleAnnotations</span> <span class='kw'>=</span> <span class='no'>sample.annotation.df</span>,
  <span class='kw'>threads</span><span class='kw'>=</span><span class='fl'>4</span>)
<span class='co'># }</span></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      
      <li><a href="#value">Value</a></li>
      
      <li><a href="#examples">Examples</a></li>
    </ul>

  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by David Jenkins, Tyler Faits, Mohammed Muzamil Khan, Emma Briars, Sebastian Carrasco Pro, W. Evan Johnson.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
   </div>

  

  </body>
</html>

