<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Convert Gene IDs — convertGeneIDs • singleCellTK</title>

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.3.7/yeti/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- sticky kit -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



<meta property="og:title" content="Convert Gene IDs — convertGeneIDs" />

<meta property="og:description" content="Convert the gene IDs in a SingleCellExperiment object using Bioconductor
org.*.eg.db data packages. Because annotation databases do not have a 1:1
relationship, this tool removes rows with no corresponding annotation in
your desired annotation, and remove any duplicate annotations after
conversion." />
<meta name="twitter:card" content="summary" />



<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">singleCellTK</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.5.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../articles/v01-Introduction_to_singleCellTK.html">Introduction to singleCellTK</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Interactive Shiny Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/v03-tab01_Upload.html">1: Upload</a>
    </li>
    <li>
      <a href="../articles/v04-tab02_Data-Summary-and-Filtering.html">2: Data Summary and Filtering</a>
    </li>
    <li>
      <a href="../articles/v05-tab03_Visualization-and-Clustering.html">3: Visualization and Clustering</a>
    </li>
    <li>
      <a href="../articles/v06-tab04_Batch-Correction.html">4: Batch Correction</a>
    </li>
    <li>
      <a href="../articles/v07-tab05_Differential-Expression.html">5: Differential Expression</a>
    </li>
    <li>
      <a href="../articles/v08-tab06_Pathway-Activity-Analysis.html">6: Pathway Activity Analysis</a>
    </li>
    <li>
      <a href="../articles/v09-tab07_Sample-Size.html">7: Sample Size</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Command Line Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/v02-Processing_and_Visualizing_Data_in_the_SingleCellTK.html">Processing and Visualizing Data</a>
    </li>
    <li>
      <a href="../articles/v10-Aligning_and_Quantifying_scRNA-Seq_Data.html">Aligning and Quantifying scRNA-Seq Data</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/compbiomed/singleCellTK">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Convert Gene IDs</h1>
    
    <div class="hidden name"><code>convertGeneIDs.Rd</code></div>
    </div>

    <div class="ref-description">
    
    <p>Convert the gene IDs in a SingleCellExperiment object using Bioconductor
org.*.eg.db data packages. Because annotation databases do not have a 1:1
relationship, this tool removes rows with no corresponding annotation in
your desired annotation, and remove any duplicate annotations after
conversion.</p>
    
    </div>

    <pre class="usage"><span class='fu'>convertGeneIDs</span>(<span class='no'>inSCE</span>, <span class='no'>inSymbol</span>, <span class='no'>outSymbol</span>, <span class='kw'>database</span> <span class='kw'>=</span> <span class='st'>"org.Hs.eg.db"</span>)</pre>
    
    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>inSCE</th>
      <td><p>Input SCtkExperiment object. Required</p></td>
    </tr>
    <tr>
      <th>inSymbol</th>
      <td><p>The input symbol type</p></td>
    </tr>
    <tr>
      <th>outSymbol</th>
      <td><p>The output symbol type</p></td>
    </tr>
    <tr>
      <th>database</th>
      <td><p>The org.*.eg.db database to use. The default is org.Hs.eg.db</p></td>
    </tr>
    </table>
    
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A SCtkExperiment with converted gene IDs.</p>
    

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='kw'>if</span>(<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/ns-load'>requireNamespace</a></span>(<span class='st'>"org.Mm.eg.db"</span>, <span class='kw'>quietly</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>)) {
  <span class='co'>#convert mouse gene symbols to ensembl IDs</span>
  <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/library'>library</a></span>(<span class='st'>"org.Mm.eg.db"</span>)
  <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/sample'>sample</a></span>(<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/colnames'>rownames</a></span>(<span class='no'>mouseBrainSubsetSCE</span>), <span class='fl'>50</span>)
  <span class='no'>mouseBrainSubsetSymbol</span> <span class='kw'>&lt;-</span> <span class='fu'>convertGeneIDs</span>(<span class='kw'>inSCE</span> <span class='kw'>=</span> <span class='no'>mouseBrainSubsetSCE</span>,
                                           <span class='kw'>inSymbol</span> <span class='kw'>=</span> <span class='st'>"SYMBOL"</span>,
                                           <span class='kw'>outSymbol</span> <span class='kw'>=</span> <span class='st'>"ENSEMBL"</span>,
                                           <span class='kw'>database</span> <span class='kw'>=</span> <span class='st'>"org.Mm.eg.db"</span>)
  <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/sample'>sample</a></span>(<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/colnames'>rownames</a></span>(<span class='no'>mouseBrainSubsetSymbol</span>), <span class='fl'>50</span>)
}</div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>Loading required package: AnnotationDbi</span></div><div class='output co'>#&gt; <span class='message'>'select()' returned 1:many mapping between keys and columns</span></div><div class='output co'>#&gt; <span class='message'>1832 ENSEMBL are NA after conversion. Removing</span></div><div class='output co'>#&gt; <span class='message'>36 ENSEMBL are duplicated after conversion. Removing additional copies</span></div><div class='output co'>#&gt; <span class='message'>299 SYMBOL are duplicated after conversion. Removing additional copies</span></div><div class='output co'>#&gt; <span class='message'>19972 SYMBOL originally, 18106 ENSEMBLs</span></div><div class='output co'>#&gt;  [1] "ENSMUSG00000070421" "ENSMUSG00000036412" "ENSMUSG00000022311"
#&gt;  [4] "ENSMUSG00000049576" "ENSMUSG00000025481" "ENSMUSG00000027244"
#&gt;  [7] "ENSMUSG00000031862" "ENSMUSG00000022636" "ENSMUSG00000029627"
#&gt; [10] "ENSMUSG00000013367" "ENSMUSG00000060780" "ENSMUSG00000039838"
#&gt; [13] "ENSMUSG00000103037" "ENSMUSG00000036172" "ENSMUSG00000028278"
#&gt; [16] "ENSMUSG00000041915" "ENSMUSG00000017950" "ENSMUSG00000032496"
#&gt; [19] "ENSMUSG00000057982" "ENSMUSG00000024370" "ENSMUSG00000029223"
#&gt; [22] "ENSMUSG00000043702" "ENSMUSG00000027270" "ENSMUSG00000027824"
#&gt; [25] "ENSMUSG00000030008" "ENSMUSG00000034751" "ENSMUSG00000021257"
#&gt; [28] "ENSMUSG00000072492" "ENSMUSG00000094230" "ENSMUSG00000057497"
#&gt; [31] "ENSMUSG00000079625" "ENSMUSG00000049680" "ENSMUSG00000056824"
#&gt; [34] "ENSMUSG00000027695" "ENSMUSG00000026404" "ENSMUSG00000045594"
#&gt; [37] "ENSMUSG00000041954" "ENSMUSG00000022235" "ENSMUSG00000054770"
#&gt; [40] "ENSMUSG00000034245" "ENSMUSG00000079434" "ENSMUSG00000011305"
#&gt; [43] "ENSMUSG00000042109" "ENSMUSG00000035354" "ENSMUSG00000018340"
#&gt; [46] "ENSMUSG00000029390" "ENSMUSG00000026123" "ENSMUSG00000007815"
#&gt; [49] "ENSMUSG00000015377" "ENSMUSG00000014245"</div><div class='input'>
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      
      <li><a href="#value">Value</a></li>
      
      <li><a href="#examples">Examples</a></li>
    </ul>

  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by David Jenkins, Tyler Faits, Mohammed Muzamil Khan, Emma Briars, Sebastian Carrasco Pro, W. Evan Johnson.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
   </div>

  

  </body>
</html>

