<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Run GSVA analysis on a SCtkExperiment object. — gsvaSCE • singleCellTK</title>

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.3.7/yeti/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- sticky kit -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



<meta property="og:title" content="Run GSVA analysis on a SCtkExperiment object. — gsvaSCE" />

<meta property="og:description" content="Run GSVA analysis on a SCtkExperiment object." />
<meta name="twitter:card" content="summary" />



<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">singleCellTK</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.5.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../articles/v01-Introduction_to_singleCellTK.html">Introduction to singleCellTK</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Interactive Shiny Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/v03-tab01_Upload.html">1: Upload</a>
    </li>
    <li>
      <a href="../articles/v04-tab02_Data-Summary-and-Filtering.html">2: Data Summary and Filtering</a>
    </li>
    <li>
      <a href="../articles/v05-tab03_Visualization-and-Clustering.html">3: Visualization and Clustering</a>
    </li>
    <li>
      <a href="../articles/v06-tab04_Batch-Correction.html">4: Batch Correction</a>
    </li>
    <li>
      <a href="../articles/v07-tab05_Differential-Expression.html">5: Differential Expression</a>
    </li>
    <li>
      <a href="../articles/v08-tab06_Pathway-Activity-Analysis.html">6: Pathway Activity Analysis</a>
    </li>
    <li>
      <a href="../articles/v09-tab07_Sample-Size.html">7: Sample Size</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Command Line Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/v02-Processing_and_Visualizing_Data_in_the_SingleCellTK.html">Processing and Visualizing Data</a>
    </li>
    <li>
      <a href="../articles/v10-Aligning_and_Quantifying_scRNA-Seq_Data.html">Aligning and Quantifying scRNA-Seq Data</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/compbiomed/singleCellTK">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Run GSVA analysis on a SCtkExperiment object.</h1>
    
    <div class="hidden name"><code>gsvaSCE.Rd</code></div>
    </div>

    <div class="ref-description">
    
    <p>Run GSVA analysis on a SCtkExperiment object.</p>
    
    </div>

    <pre class="usage"><span class='fu'>gsvaSCE</span>(<span class='no'>inSCE</span>, <span class='kw'>useAssay</span> <span class='kw'>=</span> <span class='st'>"logcounts"</span>, <span class='no'>pathwaySource</span>, <span class='no'>pathwayNames</span>, <span class='no'>...</span>)

<span class='fu'>gsvaPlot</span>(<span class='no'>inSCE</span>, <span class='no'>gsvaData</span>, <span class='no'>plotType</span>, <span class='kw'>condition</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>show_column_names</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='kw'>show_row_names</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='kw'>text_size</span> <span class='kw'>=</span> <span class='fl'>12</span>)</pre>
    
    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>inSCE</th>
      <td><p>Input SCtkExperiment object. Required</p></td>
    </tr>
    <tr>
      <th>useAssay</th>
      <td><p>Indicate which assay to use. The default is "logcounts"</p></td>
    </tr>
    <tr>
      <th>pathwaySource</th>
      <td><p>The pathway source if "Manual Input", the pathwayNames
should be rowData annotations that are (0,1) vectors. If, "MSigDB c2 (Human,
Entrez ID only)", the pathwayNames should be pathways from MSigDB c2 or "ALL"
to run on all available pathways.</p></td>
    </tr>
    <tr>
      <th>pathwayNames</th>
      <td><p>List of pathway names to run, depending on pathwaySource
parameter.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>Parameters to pass to gsva()</p></td>
    </tr>
    <tr>
      <th>gsvaData</th>
      <td><p>GSVA data to plot. Required.</p></td>
    </tr>
    <tr>
      <th>plotType</th>
      <td><p>The type of plot to use, "Violin" or "Heatmap". Required.</p></td>
    </tr>
    <tr>
      <th>condition</th>
      <td><p>The condition(s) to use for the Violin plot, or the
condition(s) to add as color bars above the Heatmap. Required for Violin,
optional for Heatmap.</p></td>
    </tr>
    <tr>
      <th>show_column_names</th>
      <td><p>Display the column labels on the heatmap. The
default is TRUE</p></td>
    </tr>
    <tr>
      <th>show_row_names</th>
      <td><p>Display the row labels on the heatmap. The default
is TRUE.</p></td>
    </tr>
    <tr>
      <th>text_size</th>
      <td><p>Text size for plots. The default is 12</p></td>
    </tr>
    </table>
    
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>gsvaSCE(): A data.frame of pathway activity scores from GSVA.</p>
<p>gsvaPlot(): The requested plot of the GSVA results.</p>
    
    <h2 class="hasAnchor" id="functions"><a class="anchor" href="#functions"></a>Functions</h2>

    <ul>
<li><p><code>gsvaPlot</code>: Plot GSVA results.</p>
<p>Plot GSVA Results</p></li>
</ul>
    

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='kw pkg'>utils</span><span class='kw ns'>::</span><span class='fu'><a href='https://www.rdocumentation.org/packages/utils/topics/data'>data</a></span>(<span class='no'>maits</span>, <span class='kw'>package</span> <span class='kw'>=</span> <span class='st'>"MAST"</span>)
<span class='kw pkg'>utils</span><span class='kw ns'>::</span><span class='fu'><a href='https://www.rdocumentation.org/packages/utils/topics/data'>data</a></span>(<span class='no'>c2BroadSets</span>, <span class='kw'>package</span> <span class='kw'>=</span> <span class='st'>"GSVAdata"</span>)
<span class='no'>maitslogtpm</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/t'>t</a></span>(<span class='no'>maits</span>$<span class='no'>expressionmat</span>)
<span class='no'>genesToSubset</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/colnames'>rownames</a></span>(<span class='no'>maitslogtpm</span>)[<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/which'>which</a></span>(<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/colnames'>rownames</a></span>(<span class='no'>maitslogtpm</span>) <span class='kw'>%in%</span>
                 <span class='kw pkg'>GSEABase</span><span class='kw ns'>::</span><span class='fu'><a href='https://www.rdocumentation.org/packages/GSEABase/topics/GeneSet-class'>geneIds</a></span>(<span class='no'>c2BroadSets</span><span class='kw'>[[</span><span class='st'>"KEGG_PROTEASOME"</span>]]))]
<span class='no'>maitslogtpm</span> <span class='kw'>&lt;-</span> <span class='no'>maitslogtpm</span>[<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/colnames'>rownames</a></span>(<span class='no'>maitslogtpm</span>) <span class='kw'>%in%</span> <span class='no'>genesToSubset</span>, ]
<span class='no'>maitsfeatures</span> <span class='kw'>&lt;-</span> <span class='no'>maits</span>$<span class='no'>fdat</span>[<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/colnames'>rownames</a></span>(<span class='no'>maits</span>$<span class='no'>fdat</span>) <span class='kw'>%in%</span> <span class='no'>genesToSubset</span>, ]
<span class='no'>maitsSCE</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='createSCE.html'>createSCE</a></span>(<span class='kw'>assayFile</span> <span class='kw'>=</span> <span class='no'>maitslogtpm</span>, <span class='kw'>annotFile</span> <span class='kw'>=</span> <span class='no'>maits</span>$<span class='no'>cdat</span>,
                      <span class='kw'>featureFile</span> <span class='kw'>=</span> <span class='no'>maitsfeatures</span>, <span class='kw'>assayName</span> <span class='kw'>=</span> <span class='st'>"logtpm"</span>,
                      <span class='kw'>inputDataFrames</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='kw'>createLogCounts</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>)
<span class='fu'>rowData</span>(<span class='no'>maitsSCE</span>)$<span class='no'>testbiomarker</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/rep'>rep</a></span>(<span class='fl'>1</span>, <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/nrow'>nrow</a></span>(<span class='no'>maitsSCE</span>))
<span class='no'>res</span> <span class='kw'>&lt;-</span> <span class='fu'>gsvaSCE</span>(<span class='kw'>inSCE</span> <span class='kw'>=</span> <span class='no'>maitsSCE</span>, <span class='kw'>useAssay</span> <span class='kw'>=</span> <span class='st'>"logtpm"</span>,
               <span class='kw'>pathwaySource</span> <span class='kw'>=</span> <span class='st'>"Manual Input"</span>, <span class='kw'>pathwayNames</span> <span class='kw'>=</span> <span class='st'>"testbiomarker"</span>,
               <span class='kw'>parallel.sz</span> <span class='kw'>=</span> <span class='fl'>1</span>)</div><div class='output co'>#&gt; Estimating GSVA scores for 1 gene sets.
#&gt; Computing observed enrichment scores
#&gt; Estimating ECDFs with Gaussian kernels
#&gt; Using parallel with 1 cores
#&gt;   |                                                                              |                                                                      |   0%  |                                                                              |======================================================================| 100%</div><div class='input'><span class='co'>#Create a small example to run</span>
<span class='kw pkg'>utils</span><span class='kw ns'>::</span><span class='fu'><a href='https://www.rdocumentation.org/packages/utils/topics/data'>data</a></span>(<span class='no'>maits</span>, <span class='kw'>package</span> <span class='kw'>=</span> <span class='st'>"MAST"</span>)
<span class='kw pkg'>utils</span><span class='kw ns'>::</span><span class='fu'><a href='https://www.rdocumentation.org/packages/utils/topics/data'>data</a></span>(<span class='no'>c2BroadSets</span>, <span class='kw'>package</span> <span class='kw'>=</span> <span class='st'>"GSVAdata"</span>)
<span class='no'>maitslogtpm</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/t'>t</a></span>(<span class='no'>maits</span>$<span class='no'>expressionmat</span>)
<span class='no'>genesToSubset</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/colnames'>rownames</a></span>(<span class='no'>maitslogtpm</span>)[<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/which'>which</a></span>(<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/colnames'>rownames</a></span>(<span class='no'>maitslogtpm</span>) <span class='kw'>%in%</span>
                 <span class='kw pkg'>GSEABase</span><span class='kw ns'>::</span><span class='fu'><a href='https://www.rdocumentation.org/packages/GSEABase/topics/GeneSet-class'>geneIds</a></span>(<span class='no'>c2BroadSets</span><span class='kw'>[[</span><span class='st'>"KEGG_PROTEASOME"</span>]]))]
<span class='no'>maitslogtpm</span> <span class='kw'>&lt;-</span> <span class='no'>maitslogtpm</span>[<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/colnames'>rownames</a></span>(<span class='no'>maitslogtpm</span>) <span class='kw'>%in%</span> <span class='no'>genesToSubset</span>, ]
<span class='no'>maitsfeatures</span> <span class='kw'>&lt;-</span> <span class='no'>maits</span>$<span class='no'>fdat</span>[<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/colnames'>rownames</a></span>(<span class='no'>maits</span>$<span class='no'>fdat</span>) <span class='kw'>%in%</span> <span class='no'>genesToSubset</span>, ]
<span class='no'>maitsSCE</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='createSCE.html'>createSCE</a></span>(<span class='kw'>assayFile</span> <span class='kw'>=</span> <span class='no'>maitslogtpm</span>, <span class='kw'>annotFile</span> <span class='kw'>=</span> <span class='no'>maits</span>$<span class='no'>cdat</span>,
                      <span class='kw'>featureFile</span> <span class='kw'>=</span> <span class='no'>maitsfeatures</span>, <span class='kw'>assayName</span> <span class='kw'>=</span> <span class='st'>"logtpm"</span>,
                      <span class='kw'>inputDataFrames</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='kw'>createLogCounts</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>)
<span class='fu'>rowData</span>(<span class='no'>maitsSCE</span>)$<span class='no'>testbiomarker</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/rep'>rep</a></span>(<span class='fl'>1</span>, <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/nrow'>nrow</a></span>(<span class='no'>maitsSCE</span>))
<span class='no'>res</span> <span class='kw'>&lt;-</span> <span class='fu'>gsvaSCE</span>(<span class='kw'>inSCE</span> <span class='kw'>=</span> <span class='no'>maitsSCE</span>, <span class='kw'>useAssay</span> <span class='kw'>=</span> <span class='st'>"logtpm"</span>,
               <span class='kw'>pathwaySource</span> <span class='kw'>=</span> <span class='st'>"Manual Input"</span>, <span class='kw'>pathwayNames</span> <span class='kw'>=</span> <span class='st'>"testbiomarker"</span>,
               <span class='kw'>parallel.sz</span> <span class='kw'>=</span> <span class='fl'>1</span>)</div><div class='output co'>#&gt; Estimating GSVA scores for 1 gene sets.
#&gt; Computing observed enrichment scores
#&gt; Estimating ECDFs with Gaussian kernels
#&gt; Using parallel with 1 cores
#&gt;   |                                                                              |                                                                      |   0%  |                                                                              |======================================================================| 100%</div><div class='input'><span class='fu'>gsvaPlot</span>(<span class='kw'>inSCE</span> <span class='kw'>=</span> <span class='no'>maitsSCE</span>, <span class='kw'>gsvaData</span> <span class='kw'>=</span> <span class='no'>res</span>,
         <span class='kw'>plotType</span> <span class='kw'>=</span> <span class='st'>"Violin"</span>, <span class='kw'>condition</span> <span class='kw'>=</span> <span class='st'>"condition"</span>)</div><div class='img'><img src='gsvaSCE-1.png' alt='' width='700' height='433' /></div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      
      <li><a href="#value">Value</a></li>

      <li><a href="#functions">Functions</a></li>
      
      <li><a href="#examples">Examples</a></li>
    </ul>

  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by David Jenkins, Tyler Faits, Mohammed Muzamil Khan, Emma Briars, Sebastian Carrasco Pro, W. Evan Johnson.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
   </div>

  

  </body>
</html>

