<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Imports gene sets from a GMT file — importGeneSetsFromGMT • singleCellTK</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />


<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Imports gene sets from a GMT file — importGeneSetsFromGMT" />
<meta property="og:description" content="Converts a list of gene sets stored in a GMT file into a
GeneSetCollection and stores it in the metadata of the
SingleCellExperiment object. These gene sets can be used in
downstream quality control and analysis functions in singleCellTK." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">singleCellTK</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.9.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../articles/introduction_to_SCTK.html">Introduction to SingleCellToolKit</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Command Line Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/import_scRNAseq_data_as_SCE.html">Import data into SCTK</a>
    </li>
    <li>
      <a href="../articles/run_qc.html">Quality Control</a>
    </li>
    <li>
      <a href="../articles/normalization.html">Normalization</a>
    </li>
    <li>
      <a href="../articles/batch_correction.html">Batch Correction</a>
    </li>
    <li>
      <a href="../articles/variability.html">Variability</a>
    </li>
    <li>
      <a href="../articles/feature_selection.html">Feature Selection</a>
    </li>
    <li>
      <a href="../articles/differential_expression.html">Differential Expression</a>
    </li>
    <li>
      <a href="../articles/visualize_results.html">Visualize Results</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/compbiomed/singleCellTK">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Imports gene sets from a GMT file</h1>
    <small class="dont-index">Source: <a href='https://github.com/compbiomed/singleCellTK/blob/master/R/importGeneSets.R'><code>R/importGeneSets.R</code></a></small>
    <div class="hidden name"><code>importGeneSetsFromGMT.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Converts a list of gene sets stored in a GMT file into a
GeneSetCollection and stores it in the metadata of the
SingleCellExperiment object. These gene sets can be used in
downstream quality control and analysis functions in <a href='singleCellTK.html'>singleCellTK</a>.</p>
    </div>

    <pre class="usage"><span class='fu'>importGeneSetsFromGMT</span>(
  <span class='no'>inSCE</span>,
  <span class='no'>file</span>,
  <span class='kw'>collectionName</span> <span class='kw'>=</span> <span class='st'>"GeneSetCollection"</span>,
  <span class='kw'>by</span> <span class='kw'>=</span> <span class='st'>"rownames"</span>,
  <span class='kw'>sep</span> <span class='kw'>=</span> <span class='st'>"\t"</span>
)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>inSCE</th>
      <td><p>Input SingleCellExperiment object.</p></td>
    </tr>
    <tr>
      <th>file</th>
      <td><p>Character. Path to GMT file. See <a href='https://rdrr.io/pkg/GSEABase/man/getObjects.html'>getGmt</a> for
more information on reading GMT files.</p></td>
    </tr>
    <tr>
      <th>collectionName</th>
      <td><p>Character. Name of collection to add gene sets to.
If this collection already exists in <code>inSCE</code>, then these gene sets will
be added to that collection. Any gene sets within the collection with the
same name will be overwritten. Default <code>GeneSetCollection</code>.</p></td>
    </tr>
    <tr>
      <th>by</th>
      <td><p>Character, character vector, or NULL. Describes the
location within <code>inSCE</code> where the gene identifiers in
<code>geneSetList</code> should be mapped. If set to <code>"rownames"</code> then the
features will be searched for among <code><a href='https://rdrr.io/r/base/colnames.html'>rownames(inSCE)</a></code>. This can also be
set to one of the column names of <code>rowData(inSCE)</code> in which case the
gene identifies will be mapped to that column in the <code>rowData</code>
of <code>inSCE</code>. <code>by</code> can be a vector the same length as
the number of gene sets in the GMT file and the elements of the vector
can point to different locations within <code>inSCE</code>. Finally, <code>by</code>
can be <code>NULL</code>. In this case, the location of the gene identifiers 
in <code>inSCE</code> should be saved in the description (2nd column)
of the GMT file. See <a href='featureIndex.html'>featureIndex</a> for more information.
Default <code>"rownames"</code>.</p></td>
    </tr>
    <tr>
      <th>sep</th>
      <td><p>Character. Delimiter of the GMT file. Default <code>"\t"</code>.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>The gene identifiers in gene sets in the GMT file will be
mapped to the rownames of <code>inSCE</code> using the <code>by</code> parameter and 
stored in a GeneSetCollection object from package
GSEABase. This object is stored in
<code>metadata(inSCE)$sctk$genesets</code>, which can be accessed in downstream
analysis functions such as <a href='runCellQC.html'>runCellQC</a>.</p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><a href='importGeneSetsFromList.html'>importGeneSetsFromList</a> for importing from lists,
<a href='importGeneSetsFromCollection.html'>importGeneSetsFromCollection</a> for importing from
GeneSetCollection objects, and
<a href='importGeneSetsFromMSigDB.html'>importGeneSetsFromMSigDB</a> for importing MSigDB gene sets.</p></div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span>(<span class='no'>scExample</span>)

<span class='co'># GMT file containing gene symbols for a subset of human mitochondrial genes</span>
<span class='no'>gmt</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span>(<span class='st'>"extdata/mito_subset.gmt"</span>, <span class='kw'>package</span> <span class='kw'>=</span> <span class='st'>"singleCellTK"</span>)

<span class='co'># "feature_name" is the second column in the GMT file, so the ids will</span>
<span class='co'># be mapped using this column in the 'rowData' of 'sce'. This </span>
<span class='co'># could also be accomplished by setting by = "feature_name" in the </span>
<span class='co'># function call. </span>
<span class='no'>sce</span> <span class='kw'>&lt;-</span> <span class='fu'>importGeneSetsFromGMT</span>(<span class='kw'>inSCE</span> <span class='kw'>=</span> <span class='no'>sce</span>, <span class='kw'>file</span> <span class='kw'>=</span> <span class='no'>gmt</span>, <span class='kw'>by</span> <span class='kw'>=</span> <span class='kw'>NULL</span>)</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Joshua D. Campbell, David Jenkins, Tyler Faits, Mohammed Muzamil Khan, Zhe Wang, Emma Briars, Sebastian Carrasco Pro, W. Evan Johnson, Yusuke Koga, Shruthi Bandyadka, Anastasia Leshchyk, Rui Hong, Yichen Wang.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


