<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Function reference â€¢ singleCellTK</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />


<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Function reference" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-index">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">singleCellTK</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.9.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../articles/introduction_to_SCTK.html">Introduction to SingleCellToolKit</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Command Line Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/import_scRNAseq_data_as_SCE.html">Import data into SCTK</a>
    </li>
    <li>
      <a href="../articles/run_qc.html">Quality Control</a>
    </li>
    <li>
      <a href="../articles/normalization.html">Normalization</a>
    </li>
    <li>
      <a href="../articles/batch_correction.html">Batch Correction</a>
    </li>
    <li>
      <a href="../articles/variability.html">Variability</a>
    </li>
    <li>
      <a href="../articles/feature_selection.html">Feature Selection</a>
    </li>
    <li>
      <a href="../articles/differential_expression.html">Differential Expression</a>
    </li>
    <li>
      <a href="../articles/visualize_results.html">Visualize Results</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/compbiomed/singleCellTK">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="contents col-md-9">
    <div class="page-header">
      <h1>Reference</h1>
    </div>

    <table class="ref-index">

    <colgroup>
      
      <col class="alias" />
      <col class="title" />
    </colgroup>

    <tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-importing-scrna-seq-data" class="hasAnchor"><a href="#section-importing-scrna-seq-data" class="anchor"></a>Importing scRNA-seq data</h2>
          <p class="section-desc"></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="importBUStools.html">importBUStools()</a></code> </p>
        </td>
        <td><p>Construct SCE object from BUStools output</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="importCellRanger.html">importCellRanger()</a></code> <code><a href="importCellRanger.html">importCellRangerV2()</a></code> <code><a href="importCellRanger.html">importCellRangerV3()</a></code> </p>
        </td>
        <td><p>Construct SCE object from Cell Ranger output</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="importCellRangerV2Sample.html">importCellRangerV2Sample()</a></code> </p>
        </td>
        <td><p>Construct SCE object from Cell Ranger V2 output for a single sample</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="importCellRangerV3Sample.html">importCellRangerV3Sample()</a></code> </p>
        </td>
        <td><p>Construct SCE object from Cell Ranger V3 output for a single sample</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="importDropEst.html">importDropEst()</a></code> </p>
        </td>
        <td><p>Create a SingleCellExperiment Object from DropEst output</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="importExampleData.html">importExampleData()</a></code> </p>
        </td>
        <td><p>Retrieve example datasets</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="importFromFiles.html">importFromFiles()</a></code> </p>
        </td>
        <td><p>Create a SingleCellExperiment object from files</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="importGeneSetsFromCollection.html">importGeneSetsFromCollection()</a></code> </p>
        </td>
        <td><p>Imports gene sets from a GeneSetCollection object</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="importGeneSetsFromGMT.html">importGeneSetsFromGMT()</a></code> </p>
        </td>
        <td><p>Imports gene sets from a GMT file</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="importGeneSetsFromList.html">importGeneSetsFromList()</a></code> </p>
        </td>
        <td><p>Imports gene sets from a list</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="importGeneSetsFromMSigDB.html">importGeneSetsFromMSigDB()</a></code> </p>
        </td>
        <td><p>Imports gene sets from MSigDB</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="importOptimus.html">importOptimus()</a></code> </p>
        </td>
        <td><p>Construct SCE object from Optimus output</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="importSEQC.html">importSEQC()</a></code> </p>
        </td>
        <td><p>Construct SCE object from seqc output</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="importSTARsolo.html">importSTARsolo()</a></code> </p>
        </td>
        <td><p>Construct SCE object from STARsolo outputs</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-quality-control-preprocessing" class="hasAnchor"><a href="#section-quality-control-preprocessing" class="anchor"></a>Quality Control &amp; Preprocessing</h2>
          <p class="section-desc"></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="runCellQC.html">runCellQC()</a></code> </p>
        </td>
        <td><p>Perform comprehensive single cell QC</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="runDropletQC.html">runDropletQC()</a></code> </p>
        </td>
        <td><p>Perform comprehensive droplet QC</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-doublet-empty-droplet-detection" class="hasAnchor"><a href="#section-doublet-empty-droplet-detection" class="anchor"></a>Doublet/Empty Droplet Detection</h2>
          <p class="section-desc"></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="runBarcodeRankDrops.html">runBarcodeRankDrops()</a></code> </p>
        </td>
        <td><p>Identify empty droplets using <a href='https://rdrr.io/pkg/DropletUtils/man/barcodeRanks.html'>barcodeRanks</a>.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="runEmptyDrops.html">runEmptyDrops()</a></code> </p>
        </td>
        <td><p>Identify empty droplets using <a href='https://rdrr.io/pkg/DropletUtils/man/emptyDrops.html'>emptyDrops</a>.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="runBcds.html">runBcds()</a></code> </p>
        </td>
        <td><p>Find doublets/multiplets using <a href='https://rdrr.io/pkg/scds/man/bcds.html'>bcds</a>.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="runCxds.html">runCxds()</a></code> </p>
        </td>
        <td><p>Find doublets/multiplets using <a href='https://rdrr.io/pkg/scds/man/cxds.html'>cxds</a>.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="runCxdsBcdsHybrid.html">runCxdsBcdsHybrid()</a></code> </p>
        </td>
        <td><p>Find doublets/multiplets using <a href='https://rdrr.io/pkg/scds/man/cxds_bcds_hybrid.html'>cxds_bcds_hybrid</a>.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="runDoubletCells.html">runDoubletCells()</a></code> </p>
        </td>
        <td><p>Detect doublet cells using <a href='https://rdrr.io/pkg/scran/man/doubletCells.html'>doubletCells</a>.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="runDoubletFinder.html">runDoubletFinder()</a></code> </p>
        </td>
        <td><p>Generates a doublet score for each cell via doubletFinder</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="runScrublet.html">runScrublet()</a></code> </p>
        </td>
        <td><p>Find doublets using <code>scrublet</code>.</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-batch-effect-correction" class="hasAnchor"><a href="#section-batch-effect-correction" class="anchor"></a>Batch Effect Correction</h2>
          <p class="section-desc"></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="runBBKNN.html">runBBKNN()</a></code> </p>
        </td>
        <td><p>Apply BBKNN batch effect correction method to SingleCellExperiment object</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="runComBat.html">runComBat()</a></code> </p>
        </td>
        <td><p>Apply ComBat batch effect correction method to SingleCellExperiment object</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="runFastMNN.html">runFastMNN()</a></code> </p>
        </td>
        <td><p>Apply a fast version of the mutual nearest neighbors (MNN) batch effect
correction method to SingleCellExperiment object</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="runHarmony.html">runHarmony()</a></code> </p>
        </td>
        <td><p>Apply Harmony batch effect correction method to SingleCellExperiment object</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="runLIGER.html">runLIGER()</a></code> </p>
        </td>
        <td><p>Apply LIGER batch effect correction method to SingleCellExperiment object</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="runLimmaBC.html">runLimmaBC()</a></code> </p>
        </td>
        <td><p>Apply Limma's batch effect correction method to SingleCellExperiment object</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="runMNNCorrect.html">runMNNCorrect()</a></code> </p>
        </td>
        <td><p>Apply the mutual nearest neighbors (MNN) batch effect correction method to
SingleCellExperiment object</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="runSCANORAMA.html">runSCANORAMA()</a></code> </p>
        </td>
        <td><p>Apply the mutual nearest neighbors (MNN) batch effect correction method to
SingleCellExperiment object</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="runSCGEN.html">runSCGEN()</a></code> </p>
        </td>
        <td><p>Apply scGen batch effect correction method to SingleCellExperiment object</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="runSCMerge.html">runSCMerge()</a></code> </p>
        </td>
        <td><p>Apply scMerge batch effect correction method to SingleCellExperiment object</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="runSeurat3Integration.html">runSeurat3Integration()</a></code> </p>
        </td>
        <td><p>Apply Integration batch effect correction method from Seurat v3 to
SingleCellExperiment object</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="runZINBWaVE.html">runZINBWaVE()</a></code> </p>
        </td>
        <td><p>Apply ZINBWaVE Batch effect correction method to SingleCellExperiment object</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-decontamination" class="hasAnchor"><a href="#section-decontamination" class="anchor"></a>Decontamination</h2>
          <p class="section-desc"></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="runDecontX.html">runDecontX()</a></code> </p>
        </td>
        <td><p>Detecting contamination with DecontX.</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-differential-expression" class="hasAnchor"><a href="#section-differential-expression" class="anchor"></a>Differential Expression</h2>
          <p class="section-desc"></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="runDEAnalysis.html">runDEAnalysis()</a></code> </p>
        </td>
        <td><p>Perform differential expression analysis on SCE with specified method
Method supported: 'MAST', 'DESeq2', 'Limma', 'ANOVA'</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="runMAST.html">runMAST()</a></code> </p>
        </td>
        <td><p>Perform differential expression analysis on SCE with MAST</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="runDESeq2.html">runDESeq2()</a></code> </p>
        </td>
        <td><p>Perform differential expression analysis on SCE with DESeq2.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="runLimmaDE.html">runLimmaDE()</a></code> </p>
        </td>
        <td><p>Perform differential expression analysis on SCE with Limma.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="runANOVA.html">runANOVA()</a></code> </p>
        </td>
        <td><p>Perform differential expression analysis on SCE with ANOVA</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-visualization" class="hasAnchor"><a href="#section-visualization" class="anchor"></a>Visualization</h2>
          <p class="section-desc"></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="plotBatchVariance.html">plotBatchVariance()</a></code> </p>
        </td>
        <td><p>Plot the percent of the variation that is explained by batch and condition
in the data</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plotBcdsResults.html">plotBcdsResults()</a></code> </p>
        </td>
        <td><p>Plots for runBcds outputs.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plotBiomarker.html">plotBiomarker()</a></code> </p>
        </td>
        <td><p>Given a set of genes, return a ggplot of expression
values.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plotCxdsResults.html">plotCxdsResults()</a></code> </p>
        </td>
        <td><p>Plots for runCxds outputs.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plotDecontXResults.html">plotDecontXResults()</a></code> </p>
        </td>
        <td><p>Plots for runDecontX outputs.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plotDEGHeatmap.html">plotDEGHeatmap()</a></code> </p>
        </td>
        <td><p>Heatmap visualization of DEG result</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plotDEGRegression.html">plotDEGRegression()</a></code> </p>
        </td>
        <td><p>plot the linear regression to show visualize the expression the of DEGs
identified by differential expression analysis</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plotDEGViolin.html">plotDEGViolin()</a></code> </p>
        </td>
        <td><p>plot the violin plot to show visualize the expression distribution of DEGs
identified by differential expression analysis</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plotDimRed.html">plotDimRed()</a></code> </p>
        </td>
        <td><p>Plot results either on already run results of reduced dimensions data.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plotDoubletCellsResults.html">plotDoubletCellsResults()</a></code> </p>
        </td>
        <td><p>Plots for runDoubletCells outputs.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plotDoubletFinderResults.html">plotDoubletFinderResults()</a></code> </p>
        </td>
        <td><p>Plots for runDoubletFinder outputs.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plotMarkerDiffExp.html">plotMarkerDiffExp()</a></code> </p>
        </td>
        <td><p>Plot a heatmap to visualize the result of <code>findMarkerDiffExp</code></p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plotPCA.html">plotPCA()</a></code> </p>
        </td>
        <td><p>Plot PCA run data from its components.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plotRunPerCellQCResults.html">plotRunPerCellQCResults()</a></code> </p>
        </td>
        <td><p>Plots for runPerCellQC outputs.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plotScdsHybridResults.html">plotScdsHybridResults()</a></code> </p>
        </td>
        <td><p>Plots for runCxdsBcdsHybrid outputs.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plotSCEBarAssayData.html">plotSCEBarAssayData()</a></code> </p>
        </td>
        <td><p>Bar plot of assay data.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plotSCEBarColData.html">plotSCEBarColData()</a></code> </p>
        </td>
        <td><p>Bar plot of colData.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plotSCEBatchFeatureMean.html">plotSCEBatchFeatureMean()</a></code> </p>
        </td>
        <td><p>Plot mean feature value in each batch of a SingleCellExperiment object</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plotSCEDensity.html">plotSCEDensity()</a></code> </p>
        </td>
        <td><p>Density plot of any data stored in the SingleCellExperiment object.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plotSCEDensityAssayData.html">plotSCEDensityAssayData()</a></code> </p>
        </td>
        <td><p>Density plot of assay data.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plotSCEDensityColData.html">plotSCEDensityColData()</a></code> </p>
        </td>
        <td><p>Density plot of colData.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plotSCEDimReduceColData.html">plotSCEDimReduceColData()</a></code> </p>
        </td>
        <td><p>Dimension reduction plot tool for colData</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plotSCEDimReduceFeatures.html">plotSCEDimReduceFeatures()</a></code> </p>
        </td>
        <td><p>Dimension reduction plot tool for assay data</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plotSCEHeatmap.html">plotSCEHeatmap()</a></code> </p>
        </td>
        <td><p>Plot heatmap of using data stored in SingleCellExperiment Object</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plotSCEScatter.html">plotSCEScatter()</a></code> </p>
        </td>
        <td><p>Dimension reduction plot tool for all types of data</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plotSCEViolin.html">plotSCEViolin()</a></code> </p>
        </td>
        <td><p>Violin plot of any data stored in the SingleCellExperiment object.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plotSCEViolinAssayData.html">plotSCEViolinAssayData()</a></code> </p>
        </td>
        <td><p>Violin plot of assay data.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plotSCEViolinColData.html">plotSCEViolinColData()</a></code> </p>
        </td>
        <td><p>Violin plot of colData.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plotScrubletResults.html">plotScrubletResults()</a></code> </p>
        </td>
        <td><p>Plots for runScrublet outputs.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plotTSNE.html">plotTSNE()</a></code> </p>
        </td>
        <td><p>Plot t-SNE plot on dimensionality reduction data run from t-SNE method.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plotUMAP.html">plotUMAP()</a></code> </p>
        </td>
        <td><p>Plot UMAP results either on already run results or run first and then plot.</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-exporting-results" class="hasAnchor"><a href="#section-exporting-results" class="anchor"></a>Exporting Results</h2>
          <p class="section-desc"></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="exportSCE.html">exportSCE()</a></code> </p>
        </td>
        <td><p>Export data in SingleCellExperiment object</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="exportSCEtoAnnData.html">exportSCEtoAnnData()</a></code> </p>
        </td>
        <td><p>Export a <a href='https://rdrr.io/pkg/SingleCellExperiment/man/SingleCellExperiment.html'>SingleCellExperiment</a> R object as 
Python annData object</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="exportSCEtoFlatFile.html">exportSCEtoFlatFile()</a></code> </p>
        </td>
        <td><p>Export a <a href='https://rdrr.io/pkg/SingleCellExperiment/man/SingleCellExperiment.html'>SingleCellExperiment</a> object to flat text files</p></td>
      </tr>
    </tbody>
    </table>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Joshua D. Campbell, David Jenkins, Tyler Faits, Mohammed Muzamil Khan, Zhe Wang, Emma Briars, Sebastian Carrasco Pro, W. Evan Johnson, Yusuke Koga, Shruthi Bandyadka, Anastasia Leshchyk, Rui Hong, Yichen Wang.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


