<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html>
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>MAST — MAST • singleCellTK</title>

<!-- jquery -->
<script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://maxcdn.bootstrapcdn.com/bootswatch/3.3.7/yeti/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">

<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../jquery.sticky-kit.min.js"></script>
<script src="../pkgdown.js"></script>



<meta property="og:title" content="MAST — MAST" />

<meta property="og:description" content="Run and visualize MAST analysis on a SCtkExperiment object." />
<meta name="twitter:card" content="summary" />


<!-- mathjax -->
<script src='https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML'></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">singleCellTK</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">1.1.4</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../articles/v01-Introduction_to_singleCellTK.html">Introduction to singleCellTK</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Interactive Shiny Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/v03-tab01_Upload.html">1: Upload</a>
    </li>
    <li>
      <a href="../articles/v04-tab02_Data-Summary-and-Filtering.html">2: Data Summary and Filtering</a>
    </li>
    <li>
      <a href="../articles/v05-tab03_Dimensionality-Reduction-and-Clustering.html">3: Dimensionality Reduction and Clustering</a>
    </li>
    <li>
      <a href="../articles/v06-tab04_Batch-Correction.html">4: Batch Correction</a>
    </li>
    <li>
      <a href="../articles/v07-tab05_Differential-Expression.html">5: Differential Expression</a>
    </li>
    <li>
      <a href="../articles/v08-tab06_Pathway-Activity-Analysis.html">6: Pathway Activity Analysis</a>
    </li>
    <li>
      <a href="../articles/v09-tab07_Sample-Size.html">7: Sample Size</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Command Line Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/v02-Processing_and_Visualizing_Data_in_the_SingleCellTK.html">Processing and Visualizing Data</a>
    </li>
    <li>
      <a href="../articles/v10-Aligning_and_Quantifying_scRNA-Seq_Data.html">Aligning and Quantifying scRNA-Seq Data</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/compbiomed/singleCellTK">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>MAST</h1>
    
    <div class="hidden name"><code>MAST.Rd</code></div>
    </div>

    <div class="ref-description">
    
    <p>Run and visualize MAST analysis on a SCtkExperiment object.</p>
    
    </div>

    <pre class="usage"><span class='fu'>MAST</span>(<span class='no'>inSCE</span>, <span class='kw'>condition</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='kw'>interest.level</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='kw'>freqExpressed</span> <span class='kw'>=</span> <span class='fl'>0.1</span>,
  <span class='kw'>fcThreshold</span> <span class='kw'>=</span> <span class='fu'>log2</span>(<span class='fl'>1.5</span>), <span class='kw'>p.value</span> <span class='kw'>=</span> <span class='fl'>0.05</span>, <span class='kw'>useThresh</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>useAssay</span> <span class='kw'>=</span> <span class='st'>"logcounts"</span>)

<span class='fu'>thresholdGenes</span>(<span class='no'>inSCE</span>, <span class='kw'>useAssay</span> <span class='kw'>=</span> <span class='st'>"logcounts"</span>)

<span class='fu'>MASTviolin</span>(<span class='no'>inSCE</span>, <span class='kw'>useAssay</span> <span class='kw'>=</span> <span class='st'>"logcounts"</span>, <span class='no'>fcHurdleSig</span>, <span class='kw'>samplesize</span> <span class='kw'>=</span> <span class='fl'>49</span>,
  <span class='kw'>threshP</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>, <span class='no'>condition</span>)

<span class='fu'>MASTregression</span>(<span class='no'>inSCE</span>, <span class='kw'>useAssay</span> <span class='kw'>=</span> <span class='st'>"logcounts"</span>, <span class='no'>fcHurdleSig</span>, <span class='kw'>samplesize</span> <span class='kw'>=</span> <span class='fl'>49</span>,
  <span class='kw'>threshP</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>, <span class='no'>condition</span>)</pre>
    
    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>inSCE</th>
      <td><p>Input SCtkExperiment object. Required</p></td>
    </tr>
    <tr>
      <th>condition</th>
      <td><p>select variable (from the colData) that is used for the model.</p></td>
    </tr>
    <tr>
      <th>interest.level</th>
      <td><p>If the condition of interest has more than two factors,
indicate which level should be used to compare to all other samples.</p></td>
    </tr>
    <tr>
      <th>freqExpressed</th>
      <td><p>Filter genes that are expressed in at least this
fraction of cells. The default is expression in 0.1 of samples.</p></td>
    </tr>
    <tr>
      <th>fcThreshold</th>
      <td><p>Minimum fold change for differentially expressed gene.</p></td>
    </tr>
    <tr>
      <th>p.value</th>
      <td><p>p values for selecting the hurdle result, default is 0.05</p></td>
    </tr>
    <tr>
      <th>useThresh</th>
      <td><p>Use adaptive thresholding to filter genes. The default is
FALSE.</p></td>
    </tr>
    <tr>
      <th>useAssay</th>
      <td><p>The assay to use for the MAST calculations. The default is
"logcounts"</p></td>
    </tr>
    <tr>
      <th>fcHurdleSig</th>
      <td><p>The filtered result from hurdle model</p></td>
    </tr>
    <tr>
      <th>samplesize</th>
      <td><p>The number of most significant genes</p></td>
    </tr>
    <tr>
      <th>threshP</th>
      <td><p>Plot threshold values from adaptive thresholding. Default is
FALSE</p></td>
    </tr>
    </table>
    
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>MAST(): A data.frame of differentially expressed genes with p-values.</p>
<p>thresholdGenes(): list of thresholded counts (on natural scale),
thresholds, bins, densities estimated on each bin, and the original data from
MAST::thresholdSCRNACountMatrix</p>
<p>MASTviolin(): A ggplot object of MAST violin plots.</p>
<p>MASTregression(): A ggplot object of MAST linear regression plots.</p>
    
    <h2 class="hasAnchor" id="functions"><a class="anchor" href="#functions"></a>Functions</h2>

    <ul>
<li><p><code>MAST</code>: Run MAST analysis.</p></li>
<li><p><code>thresholdGenes</code>: Identify adaptive thresholds</p></li>
<li><p><code>MASTviolin</code>: Visualize MAST results using violin plots</p></li>
<li><p><code>MASTregression</code>: Visualize MAST results using linear model plots</p></li>
</ul>
    

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='fu'>data</span>(<span class='st'>"mouseBrainSubsetSCE"</span>)
<span class='no'>res</span> <span class='kw'>&lt;-</span> <span class='fu'>thresholdGenes</span>(<span class='no'>mouseBrainSubsetSCE</span>)</div><div class='output co'>#&gt; <span class='message'>`fData` has no primerid.  I'll make something up.</span></div><div class='output co'>#&gt; <span class='message'>`cData` has no wellKey.  I'll make something up.</span></div><div class='output co'>#&gt; <span class='message'>Assuming data assay in position 1, with name et is log-transformed.</span></div><div class='output co'>#&gt;    (1,1.17] (1.17,1.35] (1.35,1.55] (1.55,1.76] (1.76,1.99] (1.99,2.24] 
#&gt;    1.302357    1.302357    1.302357    1.302357    1.302357    1.302357 
#&gt; (2.24,2.51]  (2.51,2.8]  (2.8,3.12] (3.12,3.47] (3.47,3.84] (3.84,8.98] 
#&gt;    1.302357    1.302357    1.302357    1.302357    1.302357    1.828119 </div><div class='input'>
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      
      <li><a href="#value">Value</a></li>

      <li><a href="#functions">Functions</a></li>
      
      <li><a href="#examples">Examples</a></li>
    </ul>

  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by David Jenkins, Tyler Faits, Emma Briars, Sebastian Carrasco Pro, W. Evan Johnson.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
   </div>

  

  </body>
</html>

