<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Find doublets using <code>scrublet</code>. — runScrublet • singleCellTK</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />


<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Find doublets using <code>scrublet</code>. — runScrublet" />
<meta property="og:description" content="A wrapper function that calls scrub_doublets from python
 module scrublet. Simulates doublets from the observed data and uses
 a k-nearest-neighbor classifier to calculate a continuous
 scrublet_score (between 0 and 1) for each transcriptome. The score
 is automatically thresholded to generate scrublet_call, a boolean
 array that is TRUE for predicted doublets and FALSE
 otherwise." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">singleCellTK</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.9.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../articles/introduction_to_SCTK.html">Introduction to SingleCellToolKit</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Command Line Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/import_scRNAseq_data_as_SCE.html">Import data into SCTK</a>
    </li>
    <li>
      <a href="../articles/run_qc.html">Quality Control</a>
    </li>
    <li>
      <a href="../articles/normalization.html">Normalization</a>
    </li>
    <li>
      <a href="../articles/batch_correction.html">Batch Correction</a>
    </li>
    <li>
      <a href="../articles/variability.html">Variability</a>
    </li>
    <li>
      <a href="../articles/feature_selection.html">Feature Selection</a>
    </li>
    <li>
      <a href="../articles/differential_expression.html">Differential Expression</a>
    </li>
    <li>
      <a href="../articles/visualize_results.html">Visualize Results</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/compbiomed/singleCellTK">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Find doublets using <code>scrublet</code>.</h1>
    <small class="dont-index">Source: <a href='https://github.com/compbiomed/singleCellTK/blob/master/R/scrublet_doubletDetection.R'><code>R/scrublet_doubletDetection.R</code></a></small>
    <div class="hidden name"><code>runScrublet.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>A wrapper function that calls <code>scrub_doublets</code> from python
 module <code>scrublet</code>. Simulates doublets from the observed data and uses
 a k-nearest-neighbor classifier to calculate a continuous
 <code>scrublet_score</code> (between 0 and 1) for each transcriptome. The score
 is automatically thresholded to generate <code>scrublet_call</code>, a boolean
 array that is <code>TRUE</code> for predicted doublets and <code>FALSE</code>
 otherwise.</p>
    </div>

    <pre class="usage"><span class='fu'>runScrublet</span>(
  <span class='no'>inSCE</span>,
  <span class='kw'>sample</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>useAssay</span> <span class='kw'>=</span> <span class='st'>"counts"</span>,
  <span class='kw'>simDoubletRatio</span> <span class='kw'>=</span> <span class='fl'>2</span>,
  <span class='kw'>nNeighbors</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>minDist</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>expectedDoubletRate</span> <span class='kw'>=</span> <span class='fl'>0.1</span>,
  <span class='kw'>stdevDoubletRate</span> <span class='kw'>=</span> <span class='fl'>0.02</span>,
  <span class='kw'>syntheticDoubletUmiSubsampling</span> <span class='kw'>=</span> <span class='fl'>1</span>,
  <span class='kw'>useApproxNeighbors</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>,
  <span class='kw'>distanceMetric</span> <span class='kw'>=</span> <span class='st'>"euclidean"</span>,
  <span class='kw'>getDoubletNeighborParents</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>minCounts</span> <span class='kw'>=</span> <span class='fl'>3</span>,
  <span class='kw'>minCells</span> <span class='kw'>=</span> <span class='fl'>3L</span>,
  <span class='kw'>minGeneVariabilityPctl</span> <span class='kw'>=</span> <span class='fl'>85</span>,
  <span class='kw'>logTransform</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>meanCenter</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>,
  <span class='kw'>normalizeVariance</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>,
  <span class='kw'>nPrinComps</span> <span class='kw'>=</span> <span class='fl'>30L</span>,
  <span class='kw'>tsneAngle</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>tsnePerplexity</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>verbose</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>,
  <span class='kw'>seed</span> <span class='kw'>=</span> <span class='fl'>12345</span>
)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>inSCE</th>
      <td><p>A <a href='https://rdrr.io/pkg/SingleCellExperiment/man/SingleCellExperiment.html'>SingleCellExperiment</a> object.
Needs <code>counts</code> in assays slot.</p></td>
    </tr>
    <tr>
      <th>sample</th>
      <td><p>Character vector. Indicates which sample each cell belongs to.
Scrublet will be run on cells from each sample separately. If NULL, then
all cells will be processed together. Default <code>NULL</code>.</p></td>
    </tr>
    <tr>
      <th>useAssay</th>
      <td><p>A string specifying which assay in the SCE to use. Default
'counts'.</p></td>
    </tr>
    <tr>
      <th>simDoubletRatio</th>
      <td><p>Numeric. Number of doublets to simulate relative to
the number of observed transcriptomes. Default 2.0.</p></td>
    </tr>
    <tr>
      <th>nNeighbors</th>
      <td><p>Integer. Number of neighbors used to construct the KNN
graph of observed transcriptomes and simulated doublets. If <code>NULL</code>,
this is set to <code><a href='https://rdrr.io/r/base/Round.html'>round(0.5 * sqrt(n_cells))</a></code>. Default <code>NULL</code>.</p></td>
    </tr>
    <tr>
      <th>minDist</th>
      <td><p>Float Determines how tightly UMAP packs points together. If <code>NULL</code>,
this is set to 0.1. Default <code>NULL</code>.</p></td>
    </tr>
    <tr>
      <th>expectedDoubletRate</th>
      <td><p>The estimated doublet rate for the experiment.
Default 0.1.</p></td>
    </tr>
    <tr>
      <th>stdevDoubletRate</th>
      <td><p>Uncertainty in the expected doublet rate.
Default 0.02.</p></td>
    </tr>
    <tr>
      <th>syntheticDoubletUmiSubsampling</th>
      <td><p>Numeric. Rate for sampling UMIs
when creating synthetic doublets. If 1.0, each doublet is created by simply
adding the UMIs from two randomly sampled observed transcriptomes. For
values less than 1, the UMI counts are added and then randomly sampled at
the specified rate. Defuault: 1.0.</p></td>
    </tr>
    <tr>
      <th>useApproxNeighbors</th>
      <td><p>Boolean. Use approximate nearest neighbor method
(<code>annoy</code>) for the KNN classifier. Default <code>TRUE</code>.</p></td>
    </tr>
    <tr>
      <th>distanceMetric</th>
      <td><p>Character. Distance metric used when finding nearest
neighbors.
For list of valid values, see the documentation for <code>annoy</code> (if
<code>useApproxNeighbors</code> is <code>TRUE</code>) or
<code>sklearn.neighbors.NearestNeighbors</code> (if <code>useApproxNeighbors</code> is
<code>FALSE</code>). Default "euclidean".</p></td>
    </tr>
    <tr>
      <th>getDoubletNeighborParents</th>
      <td><p>Boolean. If <code>TRUE</code>, return the
parent transcriptomes that generated the doublet neighbors of each
observed transcriptome. This information can be used to infer the cell
states that generated a given doublet state. Default <code>FALSE</code>.</p></td>
    </tr>
    <tr>
      <th>minCounts</th>
      <td><p>Numeric. Used for gene filtering prior to PCA. Genes
expressed at fewer than <code>minCounts</code> in fewer than <code>minCells</code>
(see below) are excluded. Default 3.</p></td>
    </tr>
    <tr>
      <th>minCells</th>
      <td><p>Integer. Used for gene filtering prior to PCA. Genes
expressed at fewer than <code>minCounts</code> (see above) in fewer than
<code>minCells</code> are excluded. Default 3.</p></td>
    </tr>
    <tr>
      <th>minGeneVariabilityPctl</th>
      <td><p>Numeric. Used for gene filtering prior to
PCA. Keep the most highly variable genes (in the top
minGeneVariabilityPctl percentile), as measured by the v-statistic
(<em>Klein et al., Cell 2015</em>). Default 85.</p></td>
    </tr>
    <tr>
      <th>logTransform</th>
      <td><p>Boolean. If <code>TRUE</code>, log-transform the counts matrix
(log10(1+TPM)). <code>sklearn.decomposition.TruncatedSVD</code> will be used for
dimensionality reduction, unless <code>meanCenter</code> is <code>TRUE</code>.
Default <code>FALSE</code>.</p></td>
    </tr>
    <tr>
      <th>meanCenter</th>
      <td><p>If <code>TRUE</code>, center the data such that each gene has a
mean of 0. <code>sklearn.decomposition.PCA</code> will be used for
dimensionality reduction. Default <code>TRUE</code>.</p></td>
    </tr>
    <tr>
      <th>normalizeVariance</th>
      <td><p>Boolean. If <code>TRUE</code>, normalize the data such
that each gene has a variance of 1.
<code>sklearn.decomposition.TruncatedSVD</code> will be used for dimensionality
reduction, unless <code>meanCenter</code> is <code>TRUE</code>. Default <code>TRUE</code>.</p></td>
    </tr>
    <tr>
      <th>nPrinComps</th>
      <td><p>Integer. Number of principal components used to embed
the transcriptomes prior to k-nearest-neighbor graph construction.
Default 30.</p></td>
    </tr>
    <tr>
      <th>tsneAngle</th>
      <td><p>Float. Determines angular size of a distant node as measured 
from a point in the t-SNE plot. If default, it is set to 0.5 Default <code>NULL</code>.</p></td>
    </tr>
    <tr>
      <th>tsnePerplexity</th>
      <td><p>Integer. The number of nearest neighbors that
is used in other manifold learning algorithms.
If default, it is set to 30. Default <code>NULL</code>.</p></td>
    </tr>
    <tr>
      <th>verbose</th>
      <td><p>Boolean. If <code>TRUE</code>, print progress updates. Default
<code>TRUE</code>.</p></td>
    </tr>
    <tr>
      <th>seed</th>
      <td><p>Seed for the random number generator. Default 12345.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A <a href='https://rdrr.io/pkg/SingleCellExperiment/man/SingleCellExperiment.html'>SingleCellExperiment</a> object with
 <code>scrub_doublets</code> output appended to the
 <a href='https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html'>colData</a> slot. The columns include
 <em>scrublet_score</em> and <em>scrublet_call</em>.</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='kw'>if</span> (<span class='fl'>FALSE</span>) {
<span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span>(<span class='no'>scExample</span>, <span class='kw'>package</span> <span class='kw'>=</span> <span class='st'>"singleCellTK"</span>)
<span class='no'>sce</span> <span class='kw'>&lt;-</span> <span class='no'>sce</span>[, <span class='fu'>colData</span>(<span class='no'>sce</span>)$<span class='no'>type</span> <span class='kw'>!=</span> <span class='st'>'EmptyDroplet'</span>]
<span class='no'>sce</span> <span class='kw'>&lt;-</span> <span class='fu'>runScrublet</span>(<span class='no'>sce</span>)
}</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Joshua D. Campbell, David Jenkins, Tyler Faits, Mohammed Muzamil Khan, Zhe Wang, Emma Briars, Sebastian Carrasco Pro, W. Evan Johnson, Yusuke Koga, Shruthi Bandyadka, Anastasia Leshchyk, Rui Hong, Yichen Wang.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


